<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stephen Pederson Dame Roma Mitchell Cancer Research Laboratories Adelaide Medical School University of Adelaide" />


<title>Integration of HiC and Super Enhancer Data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ENCODE_T47D_HiC_ENCSR549MGQ</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="description.html">Workflow Description</a>
</li>
<li>
  <a href="annotation_setup.html">Annotations</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="interactions_setup.html">Setup</a>
    </li>
    <li>
      <a href="interactions_40kb.html">40kb Interactions</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Super Enhancers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="rose.html">ROSE</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Integration of HiC and Super Enhancer Data</h1>
<h4 class="author">Stephen Pederson<br>Dame Roma Mitchell Cancer Research Laboratories<br>Adelaide Medical School<br>University of Adelaide</h4>
<h4 class="date">26 November, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-11-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>ENCODE_T47D_HiC_ENCSR549MGQ/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200918code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200918)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200918code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200918)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLENCODET47DHiCENCSR549MGQtree4d8bb28af207aae0825b5d31a3387f8377d7429ctargetblank4d8bb28a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/tree/4d8bb28af207aae0825b5d31a3387f8377d7429c" target="_blank">4d8bb28</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLENCODET47DHiCENCSR549MGQtree4d8bb28af207aae0825b5d31a3387f8377d7429ctargetblank4d8bb28a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/tree/4d8bb28af207aae0825b5d31a3387f8377d7429c" target="_blank">4d8bb28</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .snakemake/
    Ignored:    data/external/H3K27AC/bam/
    Ignored:    data/external/H3K27AC/bigwig/
    Ignored:    data/external/gencode.v33lift37.annotation.gtf.gz
    Ignored:    output/MaxHiC/

Unstaged changes:
    Modified:   output/SE.gtf
    Modified:   output/enhancers.bed

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/chip_integration.Rmd</code>) and HTML (<code>docs/chip_integration.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/4d8bb28af207aae0825b5d31a3387f8377d7429c/analysis/chip_integration.Rmd" target="_blank">4d8bb28</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-26
</td>
<td>
Added more DE genes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/e120d5420b7952e0b78fa9f622d985b23d3b5857/analysis/chip_integration.Rmd" target="_blank">e120d54</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-25
</td>
<td>
Pushed on with ChIP Integration
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/e120d5420b7952e0b78fa9f622d985b23d3b5857/docs/chip_integration.html" target="_blank">e120d54</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-25
</td>
<td>
Pushed on with ChIP Integration
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/bb2628415634f3891e42653174483da42150e7af/analysis/chip_integration.Rmd" target="_blank">bb26284</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Started integrating Leila’s data
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(rtracklayer)
library(tidyverse)
library(Gviz)
library(glue)
library(Rsamtools)
library(pander)
library(scales)
library(GenomicInteractions)
library(InteractionSet)
library(magrittr)
library(plyranges)
library(goseq)</code></pre>
<pre class="r"><code>theme_set(theme_bw())
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
panderOptions(&quot;table.split.table&quot;, Inf)</code></pre>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Given the groups of peaks identified by Leila Hosseinzadeh, using ChIP seq from both GATA3 and the AR, the connection of these peaks to enhancers, promoters and DE genes was explored. The groupings were provided as:</p>
<ol style="list-style-type: decimal">
<li>GATA3 sites detected under DHT treatment only</li>
<li>Shared sites between GATA3 and AR which are gained under DHT treatment</li>
<li>AR sites which were independent of DHT treatment</li>
<li>AR sites detected under DHT treatment only …</li>
</ol>
<p>The bed files associated with these sites were all obtained using Galaxy as part of her PhD candidature.</p>
</div>
<div id="data-import" class="section level1">
<h1>Data Import</h1>
<div id="gene-level-data" class="section level2">
<h2>Gene-Level Data</h2>
<pre class="r"><code>rnaseq &lt;- here::here(&quot;data/external/RNASeq/t47d_DHT_StrippedSerum_RNASeq_topTable.tsv&quot;) %&gt;%
  read_tsv()</code></pre>
<pre class="r"><code>geneModels &lt;- here::here(&quot;output/geneModels.rds&quot;) %&gt;%
  read_rds()
geneRanges &lt;- geneModels %&gt;% 
  split(f = .$gene) %&gt;% 
  range() %&gt;% 
  unlist() %&gt;%
  sort()
geneRanges$gene &lt;- names(geneRanges)
geneRanges$symbol &lt;- geneModels$symbol[match(geneRanges$gene, geneModels$gene)]
sq &lt;- seqinfo(geneModels)
transModels &lt;- here::here(&quot;output/transModels.rds&quot;) %&gt;%
  read_rds()</code></pre>
<p>Genes were parsed as condensed models just containing all relevant exons, and the ranges defined by the start/end points of each gene were also defined for easy comparison.</p>
<p>The RNA-Seq results from Stripped Serum were also loaded. This dataset comes with some caveats, in that very few genes were detected as DE in response to DHT treatment. This was likely to be an artefact of poor rRNA removal and the biases this leaves within a dataset. In total this dataset contained 13,013 detectable genes, broken into 55 up-regulated genes, 10 down-regulated genes, with the remaining 12,948 genes considered detectable, but not differentially expressed.</p>
<p>Whilst the comparison of AR and GATA3 ChIP peaks with differential expression may yield some insights, the clarification of the potential consequences for each peak should still be able to be extrapolated from the HiC and H3K27Ac data.</p>
</div>
<div id="hic-interactions" class="section level2">
<h2>HiC Interactions</h2>
<p>Given that the different resolutions enabled by the different bin sizes will yield differing powers to detect interactions, significant (i.e. above background) interactions were loaded at 3 different resolutions: 10kb, 20kb and 40kb.</p>
<pre class="r"><code>gi40 &lt;- here::here(&quot;output/gi_40000.rds&quot;) %&gt;% read_rds()
gi20 &lt;- here::here(&quot;output/gi_20000.rds&quot;) %&gt;% read_rds()
gi10 &lt;- here::here(&quot;output/gi_10000.rds&quot;) %&gt;% read_rds()
gi &lt;- c(gi10, gi20, gi40) %&gt;%
  sort()
gi$distance &lt;- pairdist(gi)
gi$logObsExp &lt;- log2(gi$counts / gi$exp_interactions)
gi$binSize &lt;- width(gi)$first</code></pre>
<pre class="r"><code>gi %&gt;%
  anchorOne() %&gt;%
  width() %&gt;% 
  table() %&gt;% 
  enframe(name = &quot;Bin Size&quot;, value = &quot;Number of Interactions&quot;) %&gt;% 
  mutate_all(as.numeric) %&gt;%
  pander(
    caption = &quot;Breakdown of cis interactions detected for each bin size in the Encode T47D HiC data.&quot;
  )</code></pre>
<table style="width:50%;">
<caption>Breakdown of cis interactions detected for each bin size in the Encode T47D HiC data.</caption>
<colgroup>
<col width="15%" />
<col width="34%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Bin Size</th>
<th align="center">Number of Interactions</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">10,000</td>
<td align="center">2,942</td>
</tr>
<tr class="even">
<td align="center">20,000</td>
<td align="center">8,469</td>
</tr>
<tr class="odd">
<td align="center">40,000</td>
<td align="center">18,811</td>
</tr>
</tbody>
</table>
</div>
<div id="h3k27ac-data" class="section level2">
<h2>H3K27Ac Data</h2>
<pre class="r"><code>promGR &lt;- here::here(&quot;output/promoters.bed&quot;) %&gt;% 
  import.bed() %&gt;%
  `seqinfo&lt;-`(value = sq) %&gt;%
  granges()
promGR$class &lt;- &quot;Promoter&quot;
enhGR &lt;- here::here(&quot;output/enhancers.bed&quot;) %&gt;% 
  import.bed() %&gt;%
  `seqinfo&lt;-`(value = sq) %&gt;%
  granges()
enhGR$class &lt;- &quot;Enhancer&quot;
seGR &lt;- here::here(&quot;output/SE.gtf&quot;) %&gt;%
  import.gff() %&gt;%
  `seqinfo&lt;-`(value = sq) %&gt;%
  granges()
seGR$class &lt;- &quot;Super Enhancer&quot;
h3k27ac &lt;- GRangesList(
  promGR,
  enhGR,
  seGR
) %&gt;%
  unlist() %&gt;%
  sort() %&gt;% 
  setNames(paste(seqnames(.), start(.), sep = &quot;_&quot;)) %&gt;% 
  names_to_column(&quot;peak_id&quot;) %&gt;%
  setNames(paste(.$class, .$peak_id, sep = &quot;_&quot;)) %&gt;% 
  names_to_column(&quot;feature_id&quot;)
minWidth &lt;- 75</code></pre>
<p>The initial classes of H3K27Ac defined regions were parsed and it was noted that 1,217 regions were <span class="math inline">\(\leq\)</span> than 75bp (i.e. the read length). As these were not considered to be viable peaks, these were removed.</p>
<pre class="r"><code>h3k27ac &lt;- subset(h3k27ac, width &gt; minWidth) </code></pre>
<p>It should also be noted that in the previous steps, promoters were restricted to H3K27Ac peaks which overlapped the TSS of genes detected as expressed in the RNA-Seq dataset. Any peaks which overlapped the TSS/Promoter of an undetectable gene were classed as Enhancers.</p>
<pre class="r"><code>featureTotals &lt;- table(h3k27ac$class) %&gt;%
  enframe(name = &quot;class&quot;, value = &quot;total&quot;) %&gt;%
  mutate(total = as.numeric(total))</code></pre>
<pre class="r"><code>h3k27ac %&gt;%
  with(
    tibble(
      width = width,
      class = class
    )
  ) %&gt;%
  mutate(
    class = factor(class, levels = c(&quot;Promoter&quot;, &quot;Enhancer&quot;, &quot;Super Enhancer&quot;))
  ) %&gt;%
  group_by(class) %&gt;%
  summarise(
    Total = dplyr::n(),
    `Total (bp)` = sum(width),
    `Median Width` = median(width),
    Shortest = min(width),
    Longest = max(width)
  ) %&gt;%
  rename_all(str_to_title) %&gt;%
  pander(
    caption = &quot;Description of all features defined by the H3K27Ac dataset&quot;
  )</code></pre>
<table>
<caption>Description of all features defined by the H3K27Ac dataset</caption>
<colgroup>
<col width="22%" />
<col width="11%" />
<col width="17%" />
<col width="19%" />
<col width="14%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Class</th>
<th align="center">Total</th>
<th align="center">Total (Bp)</th>
<th align="center">Median Width</th>
<th align="center">Shortest</th>
<th align="center">Longest</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Promoter</td>
<td align="center">14,228</td>
<td align="center">40,899,332</td>
<td align="center">2,270</td>
<td align="center">76</td>
<td align="center">74,203</td>
</tr>
<tr class="even">
<td align="center">Enhancer</td>
<td align="center">31,501</td>
<td align="center">41,583,128</td>
<td align="center">766</td>
<td align="center">76</td>
<td align="center">52,387</td>
</tr>
<tr class="odd">
<td align="center">Super Enhancer</td>
<td align="center">1,200</td>
<td align="center">7,298,369</td>
<td align="center">3,134</td>
<td align="center">1,033</td>
<td align="center">131,297</td>
</tr>
</tbody>
</table>
<pre class="r"><code>h3k27ByGroup &lt;- list(
  # These are the peaks seen in DHT
  E2_DHT = here::here(&quot;data/external/BED/Consensus_T47D_H3K27Ac_E2_DHT_peaks.bed&quot;) %&gt;%
    import.bed() %&gt;%
    subset(seqnames %in% seqlevels(sq)) %&gt;%
    sortSeqlevels() %&gt;%
    keepSeqlevels(value = seqlevels(sq)) %&gt;%
    subset(width &gt; minWidth) %&gt;%
    c(
      subsetByOverlaps(h3k27ac, .)
    ) %&gt;%
    sort() %&gt;%
    GenomicRanges::reduce() %&gt;%
    setNames(paste(seqnames(.), start(.), sep = &quot;_&quot;)) %&gt;%
    names_to_column(&quot;peak_id&quot;),
  # Load the peaks seen in E2 &amp; separate out into those gained &amp; lost in each group
  E2 = here::here(&quot;data/external/BED/Consensus_T47D_H3K27Ac_E2_peaks.bed&quot;) %&gt;%
    import.bed() %&gt;%
    subset(seqnames %in% seqlevels(sq)) %&gt;%
    sortSeqlevels() %&gt;%
    keepSeqlevels(value = seqlevels(sq)) %&gt;%
    subset(width &gt; minWidth) %&gt;%
    c(
      subsetByOverlaps(h3k27ac, .)
    ) %&gt;%
    sort() %&gt;%
    GenomicRanges::reduce() %&gt;%
    setNames(paste(seqnames(.), start(.), sep = &quot;_&quot;)) %&gt;%
    names_to_column(&quot;peak_id&quot;)
) %&gt;%
  lapply(`seqinfo&lt;-`, value = sq)</code></pre>
<pre class="r"><code>h3k27ac$E2_DHT &lt;- h3k27ac$peak_id %in% h3k27ByGroup$E2_DHT$peak_id
h3k27ac$E2 &lt;- h3k27ac$peak_id %in% h3k27ByGroup$E2$peak_id
h3k27ac$group &lt;- mcols(h3k27ac) %&gt;%
  with(
    case_when(
      E2_DHT &amp; E2 ~ &quot;Common&quot;,
      !E2_DHT &amp; E2 ~ &quot;E2 Only&quot;,
      E2_DHT &amp; !E2 ~ &quot;DHT Only&quot;,
      !E2_DHT &amp; !E2 ~ &quot;None&quot; ## WHY DO THESE EXIST!!!!?
    )
  )</code></pre>
<p>Information regarding which dataset contained an overlap with each consensus peak was then added to each H3K27Ac site.</p>
<pre class="r"><code>h3k27ac %&gt;%
  # subset(E2_DHT | E2) %&gt;%
  mcols() %&gt;%
  as.data.frame() %&gt;%
  group_by(class, E2_DHT, E2) %&gt;%
  tally() %&gt;%
  ungroup() %&gt;%
  pivot_wider(names_from = class, values_from = n, values_fill = 0) %&gt;%
  mutate(
    `Treatment Group` = case_when(
      E2 &amp; E2_DHT ~ &quot;Both&quot;,
      !E2 &amp; E2_DHT ~ &quot;DHT Only&quot;,
      E2 &amp; !E2_DHT ~ &quot;E2 Only&quot;,
      !E2 &amp; !E2_DHT ~ &quot;None&quot;
    )
    ) %&gt;%
  arrange(`Treatment Group`) %&gt;%
  dplyr::select(`Treatment Group`, Promoter, contains(&quot;Enhancer&quot;)) %&gt;%
  pander(
    caption = &quot;
    Summary of consensus H3K27Ac features by treatment group.
    Notably, all Super Enhancers contained peaks which overlapped them in each group, however this does not ensure that all Super Enhancers were defined as Super Ehancers in both treatment groups.
    The ROSE workflow did detect a small number of SEs which were unique to DHT treatment.
    &quot;
  )</code></pre>
<table style="width:79%;">
<caption>Summary of consensus H3K27Ac features by treatment group. Notably, all Super Enhancers contained peaks which overlapped them in each group, however this does not ensure that all Super Enhancers were defined as Super Ehancers in both treatment groups. The ROSE workflow did detect a small number of SEs which were unique to DHT treatment.</caption>
<colgroup>
<col width="25%" />
<col width="15%" />
<col width="15%" />
<col width="23%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Treatment Group</th>
<th align="center">Promoter</th>
<th align="center">Enhancer</th>
<th align="center">Super Enhancer</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Both</td>
<td align="center">13,326</td>
<td align="center">21,768</td>
<td align="center">1,200</td>
</tr>
<tr class="even">
<td align="center">DHT Only</td>
<td align="center">666</td>
<td align="center">6,214</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td align="center">E2 Only</td>
<td align="center">236</td>
<td align="center">2,715</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td align="center">None</td>
<td align="center">0</td>
<td align="center">804</td>
<td align="center">0</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="all-chip-data" class="section level1">
<h1>All ChIP Data</h1>
<p>Both GATA3 and AR ChIP datasets were loaded and consensus peaks identified. These were then annotated simply based on gain or loss in DHT treatment, noting that GATA3 data contains two conditions (<em>Vehicle</em> and <em>DHT</em>) whilst AR data contained the conditions <em>E2</em> and <em>E2+DHT</em>.</p>
<div id="all-gata3-sites" class="section level2">
<h2>All GATA3 Sites</h2>
<pre class="r"><code>gata3GR &lt;- here::here(&quot;data/external/BED&quot;) %&gt;%
  list.files(pattern = &quot;GATA3.+bed&quot;, full.names = TRUE) %&gt;%
  sapply(function(x){
    import.bed(x) %&gt;%
      subset(seqnames %in% seqlevels(sq)) %&gt;%
      sortSeqlevels() %&gt;%
      keepSeqlevels(value = seqlevels(sq)) %&gt;%
      `seqinfo&lt;-`(value = sq) %&gt;%
      sort
  }, simplify = FALSE) %&gt;%
  setNames(str_extract(names(.), &quot;Vehicle|DHT&quot;)) 
gata3ConsGR &lt;- gata3GR %&gt;%
  as(&quot;GRangesList&quot;) %&gt;%
  unlist() %&gt;%
  GenomicRanges::reduce() %&gt;%
  setNames(paste0(seqnames(.), &quot;_&quot;, start(.))) %&gt;%
  names_to_column(&quot;peak_id&quot;) %&gt;%
  subset(width &gt; minWidth)
gata3ConsGR$Vehicle &lt;- seq_along(gata3ConsGR) %in% queryHits(findOverlaps(gata3ConsGR, gata3GR$Vehicle))
gata3ConsGR$DHT &lt;- seq_along(gata3ConsGR) %in% queryHits(findOverlaps(gata3ConsGR, gata3GR$DHT))
gata3ConsGR$group &lt;- mcols(gata3ConsGR) %&gt;%
  with(
    case_when(
      Vehicle &amp; DHT ~ &quot;Common&quot;,
      !Vehicle &amp; DHT ~ &quot;DHT Only&quot;,
      Vehicle &amp; !DHT ~ &quot;Vehicle Only&quot;,
      !Vehicle &amp; !DHT ~ NA_character_
    )
  )</code></pre>
<p>In order to characterise the behaviours of the defined groups, the consensus set of 92,532 GATA3 peaks was defined as those detected in either Vehicle or DHT. In keeping with all steps of this analysis, any peaks <span class="math inline">\(\leq 75\)</span>bp were excluded.</p>
<pre class="r"><code>gata3ConsGR$group %&gt;%
  table() %&gt;%
  enframe(name = &quot;Group&quot;, value = &quot;Peaks&quot;) %&gt;%
  mutate(Peaks = as.numeric(Peaks)) %&gt;%
  pander(
    caption = glue(
      &quot;Summary of consensus GATA3 peaks, after exclusion of any peaks $\\leq${minWidth}bp.&quot;
    )
  )</code></pre>
<table style="width:33%;">
<caption>Summary of consensus GATA3 peaks, after exclusion of any peaks $$75bp.</caption>
<colgroup>
<col width="20%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Group</th>
<th align="center">Peaks</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Common</td>
<td align="center">64,089</td>
</tr>
<tr class="even">
<td align="center">DHT Only</td>
<td align="center">16,364</td>
</tr>
<tr class="odd">
<td align="center">Vehicle Only</td>
<td align="center">12,079</td>
</tr>
</tbody>
</table>
<pre class="r"><code>gata3Cons2Feature &lt;- gata3ConsGR %&gt;% 
  join_overlap_inner(h3k27ac) %&gt;% 
  mcols() %&gt;% 
  split(f = .$peak_id.x) %&gt;% 
  lapply(function(x){unique(x$class)})</code></pre>
<p>Each consensus peak was then mapped to the features which overlapped the peak. A total of 20,131 peaks were mapped to 21,900 features, noting that many enhancers will overlap super enhancers.</p>
</div>
<div id="all-ar-sites" class="section level2">
<h2>All AR Sites</h2>
<pre class="r"><code>arGR &lt;- here::here(&quot;data/external/BED&quot;) %&gt;%
  list.files(pattern = &quot;AR.+.bed&quot;, full.names = TRUE) %&gt;%
  sapply(function(x){
    import.bed(x) %&gt;%
      subset(seqnames %in% seqlevels(sq)) %&gt;%
      sortSeqlevels() %&gt;%
      keepSeqlevels(value = seqlevels(sq)) %&gt;%
      `seqinfo&lt;-`(value = sq) %&gt;%
      sort() %&gt;%
      granges()
  }, simplify = FALSE) %&gt;%
  setNames(str_replace_all(names(.), &quot;.+(E2|E2_DHT)_peaks.+&quot;, &quot;\\1&quot;)) 
arConsGR &lt;- arGR %&gt;%
  as(&quot;GRangesList&quot;) %&gt;%
  unlist() %&gt;%
  GenomicRanges::reduce() %&gt;%
  setNames(paste0(seqnames(.), &quot;_&quot;, start(.))) %&gt;%
  names_to_column(&quot;peak_id&quot;) %&gt;%
  subset(width &gt; minWidth)
arConsGR$E2 &lt;- seq_along(arConsGR) %in% queryHits(findOverlaps(arConsGR, arGR$E2))
arConsGR$E2_DHT &lt;- seq_along(arConsGR) %in% queryHits(findOverlaps(arConsGR, arGR$E2_DHT))
arConsGR$group &lt;- mcols(arConsGR) %&gt;%
  with(
    case_when(
      E2 &amp; E2_DHT ~ &quot;Common&quot;,
      !E2 &amp; E2_DHT ~ &quot;E2+DHT Only&quot;,
      E2 &amp; !E2_DHT ~ &quot;E2 Only&quot;,
      !E2 &amp; !E2_DHT ~ NA_character_
    )
  )</code></pre>
<p>In order to characterise the behaviours of the defined groups, the consensus set of 27,540 AR peaks was defined as those detected in either Vehicle or DHT. In keeping with all steps of this analysis, any peaks <span class="math inline">\(\leq 75\)</span>bp were excluded.</p>
<pre class="r"><code>arConsGR$group %&gt;%
  table() %&gt;%
  enframe(name = &quot;Group&quot;, value = &quot;Peaks&quot;) %&gt;%
  mutate(Peaks = as.numeric(Peaks)) %&gt;%
  pander(
    caption = glue(
      &quot;Summary of consensus AR peaks, after exclusion of any peaks $\\leq${minWidth}bp.&quot;
    )
  )</code></pre>
<table style="width:32%;">
<caption>Summary of consensus AR peaks, after exclusion of any peaks $$75bp.</caption>
<colgroup>
<col width="19%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Group</th>
<th align="center">Peaks</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Common</td>
<td align="center">6,796</td>
</tr>
<tr class="even">
<td align="center">E2 Only</td>
<td align="center">3</td>
</tr>
<tr class="odd">
<td align="center">E2+DHT Only</td>
<td align="center">20,741</td>
</tr>
</tbody>
</table>
<pre class="r"><code>arCons2Feature &lt;- arConsGR %&gt;% 
  join_overlap_inner(h3k27ac) %&gt;% 
  mcols() %&gt;% 
  split(f = .$peak_id.x) %&gt;% 
  lapply(function(x){unique(x$class)})</code></pre>
<p>Each consensus peak was then mapped to the features which overlapped the peak. A total of 13,583 peaks were mapped to 15,218 features, noting that many enhancers will overlap super enhancers.</p>
</div>
</div>
<div id="chip-peaks-by-ar-gata3-behavioural-groups" class="section level1">
<h1>ChIP Peaks By AR &amp; GATA3 Behavioural Groups</h1>
<div id="plotting-setup" class="section level2">
<h2>Plotting Setup</h2>
<pre class="r"><code>gen &lt;- &quot;hg19&quot;
ax &lt;- GenomeAxisTrack()
plotRegion &lt;- function(gr, zoom = 1, hic = gi, h3k27 = h3k27ac, gata3 = gata3ConsGR, ar = arConsGR, gm = geneModels, .ax = ax, rna = rnaseq, shift = 0, fs = 12, highlight = TRUE){
  
  stopifnot(is(gr, &quot;GRanges&quot;) | length(gr) &gt; 0)
  
  ## Form the Ideogram track
  chr &lt;- unique(as.character(seqnames(gr)))
  ideo &lt;- IdeogramTrack(genome = gen, chromosome = chr)
  
  ## The HiC Track
  goiInteractions &lt;- hic %&gt;%
    subsetByOverlaps(gr, ignore.strand = TRUE) %&gt;%
    # subset(distance &lt; width(gr)*zoom) %&gt;%
    # subset(distance &lt; maxIntDist) %&gt;%
    as(&quot;GenomicInteractions&quot;)
  hicTR &lt;- goiInteractions %&gt;%
    InteractionTrack(
      chromosome = chr,
      name = &quot;HiC&quot;
    )
  displayPars(hicTR) &lt;- list(fontsize = fs)
  
  ## Expand the range after looking for interactions within the initial range
  ## This keeps the range of interest in the centre
  origGR &lt;- gr
  gr &lt;- resize(gr, width = zoom*width(gr), fix = &quot;center&quot;)
  gr &lt;- shift(gr, shift)

  ## Setup the promoters, enhancers and super-enhancers as separate tracks
  promTR &lt;- h3k27 %&gt;%
    subsetByOverlaps(gr, ignore.strand = TRUE) %&gt;%
    subset(class == &quot;Promoter&quot;) %&gt;%
    GeneRegionTrack(
      name = &quot;Promoter&quot;,
      stacking = &quot;dense&quot;,
      col = &quot;transparent&quot;,
      feature = .$group,
      fontsize = fs,
      `Common` = &quot;grey&quot;,
      `DHT Only` = &quot;red&quot;,
      `E2 Only` = &quot;blue&quot;
    )
  if (length(promTR) == 0) promTR &lt;- c()
  enhTR &lt;- h3k27 %&gt;%
    subsetByOverlaps(gr, ignore.strand = TRUE) %&gt;%
    subset(class == &quot;Enhancer&quot;) %&gt;%
    GeneRegionTrack(
      name = &quot;Enhancer&quot;,
      stacking = &quot;dense&quot;,
      col = &quot;transparent&quot;,
      feature = .$group,
      fontsize = fs,      
      `Common` = &quot;grey&quot;,
      `DHT Only` = &quot;red&quot;,
      `E2 Only` = &quot;blue&quot;,
      None = &quot;grey&quot;
    )
  if (length(enhTR) == 0) enhTR &lt;- c()
  seTR &lt;- h3k27 %&gt;%
    subsetByOverlaps(gr, ignore.strand = TRUE) %&gt;%
    subset(class == &quot;Super Enhancer&quot;) %&gt;%
    GeneRegionTrack(
      name = &quot;Super Enh.&quot;,
      stacking = &quot;dense&quot;,
      col = &quot;transparent&quot;,
      feature = .$group,
      fontsize = fs,      
      `Common` = &quot;grey&quot;,
      `DHT Only` = &quot;red&quot;,
      `E2 Only` = &quot;blue&quot;
    )
  if (length(seTR) == 0) seTR &lt;- c()
  
  gata3TR &lt;- gata3 %&gt;%
    subsetByOverlaps(gr, ignore.strand = TRUE) %&gt;%
    GeneRegionTrack(
      name = &quot;GATA3&quot;,
      stacking = &quot;dense&quot;,
      feature = .$group,
      size = 1,
      col = &quot;transparent&quot;,
      fontsize = fs,      
      `Common` = &quot;grey&quot;,
      `DHT Only` = &quot;red&quot;,
      `Vehicle Only` = &quot;blue&quot;
    )
  arTR &lt;- ar %&gt;%
    subsetByOverlaps(gr, ignore.strand = TRUE) %&gt;%
    GeneRegionTrack(
      name = &quot;AR&quot;,
      stacking = &quot;dense&quot;,
      feature = .$group,
      size = 1,
      col = &quot;transparent&quot;,
      fontsize = fs,      
      `Common` = &quot;grey&quot;,
      `E2+DHT Only` = &quot;red&quot;,
      `E2 Only` = &quot;blue&quot;
    )
  
  olGenes &lt;- unique(subsetByOverlaps(gm, gr, ignore.strand = TRUE)$gene)
  expTR &lt;- gm %&gt;%
    subset(gene %in% olGenes) %&gt;%
    subset(gene %in% rna$gene_id) %&gt;%
    GeneRegionTrack(
      name = &quot;Expressed&quot;,
      transcriptAnnotation = &quot;symbol&quot;,
      fill = &quot;#008000&quot;,
      col = &quot;transparent&quot;,
      fontsize = fs
    )
  notExpTR &lt;- gm %&gt;%
    subset(gene %in% olGenes) %&gt;%
    subset(!gene %in% rna$gene_id) %&gt;%
    GeneRegionTrack(
      name = &quot;Not Expressed&quot;,
      transcriptAnnotation = &quot;symbol&quot;,
      fill = rgb(0.5, 0.5, 0.5, 0.5),
      size = 0.8,
      fontsize = fs,  
      collapseTranscripts = &quot;meta&quot;
    )
  
  ## A rectangle highlight around the original range
  highTR &lt;- c(gata3TR, arTR, promTR, enhTR, seTR, expTR)
  if (highlight) {
    highTR &lt;- HighlightTrack(
      trackList = c(gata3TR, arTR, promTR, enhTR, seTR, expTR),
      range = resize(origGR, width = max(1e-2*zoom*width(origGR), width(origGR)), fix = &quot;center&quot;),
      col = rgb(0.9, 0.7, 0),
      fill = &quot;#FFFFFF00&quot;
    )  
  }

  # Now form the plot
  list(
    ideo, .ax, hicTR
  ) %&gt;%
    c(highTR) %&gt;%
    c(notExpTR) %&gt;%
    plotTracks(
      from = start(gr),
      to = end(gr)
    )
}</code></pre>
<p>As much of the group-level analysis will involve plotting, a simple wrapper function was written to enable simple plotting of all tracks.</p>
</div>
<div id="gata3-sites-gained-with-dht" class="section level2">
<h2>GATA3 Sites Gained With DHT</h2>
<pre class="r"><code>chipGR &lt;- here::here(&quot;data/external/BED/AR_GATA3_Groups/Galaxy280-[T47D-GATA3-DHT-ONLY].bed&quot;) %&gt;%
  import.bed() %&gt;%
  subset(seqnames %in% seqlevels(sq)) %&gt;%
  sortSeqlevels() %&gt;%
  keepSeqlevels(value = seqlevels(sq)) %&gt;%
  sort() %&gt;%
  subset(width &gt; minWidth) %&gt;%
  c(
    subsetByOverlaps(gata3ConsGR, .)
  ) %&gt;%
  GenomicRanges::reduce() %&gt;%
  setNames(paste0(seqnames(.), &quot;_&quot;, start(.))) %&gt;%
  names_to_column(&quot;peak_id&quot;) %&gt;%
  `seqinfo&lt;-`(value = sq)</code></pre>
<p>The first group of sites to integrate were the GATA3 ChIP peaks gained under DHT treatment, but where no corresponding increase was seen in the AR ChIP peaks. Data was imported for a total of 66,014 peaks, excluding peaks <span class="math inline">\(\leq\)</span> 75bp. This peaks were merged with the consensus GATA3 peaks defined previously to ensure direct comparability between groups. This set of peaks captured 71% of consensus GATA3 peaks.</p>
<pre class="r"><code>ol &lt;- findOverlaps(chipGR, h3k27ac)</code></pre>
<p>The first step is to subset and merge these sites with those which overlap a defined promoter or enhancer. This step reduced the set of peaks to 66,014 unique peaks, however, given that some peaks may partially overlap multiple H3K27Ac-defined regions, this represented a total of 66,014 mappings between the two datasets.</p>
<pre class="r"><code>chipGR %&gt;%
  join_overlap_inner(h3k27ac) %&gt;%
  with(
    tibble(
      width = width,
      class = class,
      feature_id = feature_id
    )
  ) %&gt;%
  group_by(class) %&gt;%
  summarise(
    `Total Mapped Peaks` = dplyr::n(),
    `Total (bp)` = sum(width),
    `Median Width` = median(width),
    Shortest = min(width),
    Longest = max(width)
  ) %&gt;%
  rename_all(str_to_title) %&gt;%
  pander(
    caption = &quot;Summary of mappings from ChIP peaks (DHT-gained GATA3 peaks) to regions defined by the H3K27Ac dataset&quot;
  )</code></pre>
<table>
<caption>Summary of mappings from ChIP peaks (DHT-gained GATA3 peaks) to regions defined by the H3K27Ac dataset</caption>
<colgroup>
<col width="19%" />
<col width="23%" />
<col width="14%" />
<col width="17%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Class</th>
<th align="center">Total Mapped Peaks</th>
<th align="center">Total (Bp)</th>
<th align="center">Median Width</th>
<th align="center">Shortest</th>
<th align="center">Longest</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Enhancer</td>
<td align="center">6,288</td>
<td align="center">2,739,720</td>
<td align="center">379</td>
<td align="center">77</td>
<td align="center">2,083</td>
</tr>
<tr class="even">
<td align="center">Promoter</td>
<td align="center">4,410</td>
<td align="center">1,821,144</td>
<td align="center">366</td>
<td align="center">77</td>
<td align="center">2,390</td>
</tr>
<tr class="odd">
<td align="center">Super Enhancer</td>
<td align="center">716</td>
<td align="center">310,429</td>
<td align="center">357</td>
<td align="center">105</td>
<td align="center">2,083</td>
</tr>
</tbody>
</table>
<div id="enrichment-testing" class="section level3">
<h3>Enrichment Testing</h3>
<p>The first step is to ascertain whether the peaks where GATA3 is gained under DHT treatment, are associated with any specific feature from the H3K27Ac dataset, such as Promoters, Enhancers of Super Enhancers. The size of the peak may bias any mapping and this was used as a sampling offset using the Wallenius approximation for biased sampling in the hypergeometric distribution.</p>
<p>For enrichment testing, each feature was considered as analogous to a gene, with a peak being mapped to it being considered as ‘success’. All 3 feature types were assessed by considering that feature type, and grouping the other feature types. In this manner, only the peaks which overlapped a defined feature were considered for enrichment testing.</p>
<pre class="r"><code>mapped &lt;- (gata3ConsGR$peak_id %in% chipGR$peak_id) %&gt;%
  setNames(gata3ConsGR$peak_id)
pwf &lt;- nullp(mapped, bias.data = width(gata3ConsGR), plot.fit = FALSE)
enrichRes &lt;- goseq(pwf, gene2cat = gata3Cons2Feature)
enrichRes %&gt;%
  mutate(
    Percentage = percent(numDEInCat / numInCat),
    over_represented_pvalue = sprintf(&quot;%.3e&quot;, p.adjust(over_represented_pvalue, &quot;bonf&quot;))
  ) %&gt;%
  dplyr::select(
    Feature = category,
    adjP = over_represented_pvalue,
    `Number Mapped in Group Peaks` = numDEInCat,
    `Consensus Peaks Mapped` = numInCat,
    `Percentage Mapped` = Percentage
  ) %&gt;%
  pander(
    caption = &quot;Assessment of enrichment for mappings of DHT-responsive GATA3 peaks to genomic features defined by H3K27Ac data&quot;
  )</code></pre>
<table>
<caption>Assessment of enrichment for mappings of DHT-responsive GATA3 peaks to genomic features defined by H3K27Ac data</caption>
<colgroup>
<col width="16%" />
<col width="12%" />
<col width="29%" />
<col width="23%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Feature</th>
<th align="center">adjP</th>
<th align="center">Number Mapped in Group Peaks</th>
<th align="center">Consensus Peaks Mapped</th>
<th align="center">Percentage Mapped</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Promoter</td>
<td align="center">1.291e-144</td>
<td align="center">4,381</td>
<td align="center">6,654</td>
<td align="center">65.8%</td>
</tr>
<tr class="even">
<td align="center">Enhancer</td>
<td align="center">1.000e+00</td>
<td align="center">6,040</td>
<td align="center">13,468</td>
<td align="center">44.8%</td>
</tr>
<tr class="odd">
<td align="center">Super Enhancer</td>
<td align="center">1.000e+00</td>
<td align="center">716</td>
<td align="center">1,778</td>
<td align="center">40.3%</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="comparison-between-promoters-and-gene-expression" class="section level2">
<h2>Comparison between Promoters and Gene Expression</h2>
<div id="differentially-expressed-genes" class="section level3 tabset">
<h3>Differentially Expressed Genes</h3>
<p>Of the 13,013 genes which were detectable in the corresponding RNA-Seq analysis, peaks corresponding to a gain of GATA3 under DHT treatment were found in the promoters of 2907 genes.</p>
<pre class="r"><code>chipGR %&gt;%
  join_overlap_intersect(h3k27ac) %&gt;%
  subset(class == &quot;Promoter&quot;) %&gt;%
  join_overlap_intersect(geneRanges) %&gt;%
  mcols() %&gt;%
  as.data.frame() %&gt;%
  as_tibble() %&gt;%
  dplyr::select(-peak_id.y) %&gt;%
  chop(c(peak_id.x, feature_id)) %&gt;%
  mutate(
    `H3K27Ac Treatment Group` = case_when(
      E2_DHT &amp; E2 ~ &quot;Both&quot;,
      !E2_DHT &amp; E2 ~ &quot;E2 Only&quot;,
      E2_DHT &amp; !E2 ~ &quot;DHT Only&quot;
    )
  ) %&gt;%
  inner_join(rnaseq, by = c(&quot;gene&quot; = &quot;gene_id&quot;, &quot;symbol&quot; = &quot;gene_name&quot;)) %&gt;%
  arrange(PValue) %&gt;%
  dplyr::filter(DE) %&gt;%
  dplyr::mutate(
    `Peaks In Promoter` = vapply(peak_id.x, length, integer(1)),
    PValue = sprintf(&quot;%.2e&quot;, PValue),
    FDR = case_when(
      FDR &gt; 0.001 ~ sprintf(&quot;%.3f&quot;, FDR),
      FDR &lt;= 0.001 ~ sprintf(&quot;%.3e&quot;, FDR)
    )
  ) %&gt;%
  dplyr::select(ID = gene, Name = symbol, `Peaks In Promoter`, `H3K27Ac Treatment Group`,  logFC, logCPM, PValue, FDR) %&gt;%
  pander(
    justify = &quot;llrlrrrr&quot;,
    caption = &quot;
    RNA-Seq results for DHT-responsive GATA3 peaks which map to promoters. 
    Only Differentially expressed genes are shown, to an FDR of 5%. 
    All but one of these genes was up-regulated, however, given that most DE genes wer up-regulated this was not statistically significant.
    &quot;,
    emphasize.strong.rows = which(str_detect(.$`H3K27Ac Treatment Group`, &quot;DHT&quot;))
  )</code></pre>
<table>
<caption>RNA-Seq results for DHT-responsive GATA3 peaks which map to promoters. Only Differentially expressed genes are shown, to an FDR of 5%. All but one of these genes was up-regulated, however, given that most DE genes wer up-regulated this was not statistically significant.</caption>
<colgroup>
<col width="16%" />
<col width="9%" />
<col width="14%" />
<col width="19%" />
<col width="8%" />
<col width="8%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">ID</th>
<th align="left">Name</th>
<th align="right">Peaks In Promoter</th>
<th align="left">H3K27Ac Treatment Group</th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000164188</td>
<td align="left">RANBP3L</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">2.343</td>
<td align="right">5.477</td>
<td align="right">1.82e-11</td>
<td align="right">7.320e-08</td>
</tr>
<tr class="even">
<td align="left">ENSG00000135373</td>
<td align="left">EHF</td>
<td align="right">5</td>
<td align="left">Both</td>
<td align="right">1.075</td>
<td align="right">7.95</td>
<td align="right">2.25e-11</td>
<td align="right">7.320e-08</td>
</tr>
<tr class="odd">
<td align="left"><strong>ENSG00000109906</strong></td>
<td align="left"><strong>ZBTB16</strong></td>
<td align="right"><strong>1</strong></td>
<td align="left"><strong>DHT Only</strong></td>
<td align="right"><strong>5.609</strong></td>
<td align="right"><strong>2.315</strong></td>
<td align="right"><strong>8.05e-10</strong></td>
<td align="right"><strong>1.494e-06</strong></td>
</tr>
<tr class="even">
<td align="left">ENSG00000152661</td>
<td align="left">GJA1</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">1.941</td>
<td align="right">2.698</td>
<td align="right">9.18e-10</td>
<td align="right">1.494e-06</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000127663</td>
<td align="left">KDM4B</td>
<td align="right">2</td>
<td align="left">Both</td>
<td align="right">1.165</td>
<td align="right">5.407</td>
<td align="right">4.53e-09</td>
<td align="right">5.892e-06</td>
</tr>
<tr class="even">
<td align="left">ENSG00000111961</td>
<td align="left">SASH1</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">1.006</td>
<td align="right">4.939</td>
<td align="right">2.18e-08</td>
<td align="right">2.360e-05</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000181744</td>
<td align="left">DIPK2A</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">0.8745</td>
<td align="right">6.018</td>
<td align="right">2.76e-08</td>
<td align="right">2.763e-05</td>
</tr>
<tr class="even">
<td align="left">ENSG00000167291</td>
<td align="left">TBC1D16</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">0.7529</td>
<td align="right">5.751</td>
<td align="right">4.99e-08</td>
<td align="right">4.634e-05</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000196208</td>
<td align="left">GREB1</td>
<td align="right">5</td>
<td align="left">Both</td>
<td align="right">0.7876</td>
<td align="right">6.165</td>
<td align="right">6.58e-08</td>
<td align="right">5.039e-05</td>
</tr>
<tr class="even">
<td align="left">ENSG00000170017</td>
<td align="left">ALCAM</td>
<td align="right">3</td>
<td align="left">Both</td>
<td align="right">0.726</td>
<td align="right">7.451</td>
<td align="right">6.75e-08</td>
<td align="right">5.039e-05</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000096060</td>
<td align="left">FKBP5</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">0.981</td>
<td align="right">6.455</td>
<td align="right">6.97e-08</td>
<td align="right">5.039e-05</td>
</tr>
<tr class="even">
<td align="left">ENSG00000179104</td>
<td align="left">TMTC2</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">0.7113</td>
<td align="right">5.622</td>
<td align="right">4.88e-07</td>
<td align="right">2.889e-04</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000143153</td>
<td align="left">ATP1B1</td>
<td align="right">2</td>
<td align="left">Both</td>
<td align="right">0.6196</td>
<td align="right">6.459</td>
<td align="right">1.13e-06</td>
<td align="right">6.110e-04</td>
</tr>
<tr class="even">
<td align="left">ENSG00000064042</td>
<td align="left">LIMCH1</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">0.5892</td>
<td align="right">5.103</td>
<td align="right">4.80e-06</td>
<td align="right">0.002</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000137266</td>
<td align="left">SLC22A23</td>
<td align="right">3</td>
<td align="left">Both</td>
<td align="right">0.6319</td>
<td align="right">5.852</td>
<td align="right">1.34e-05</td>
<td align="right">0.005</td>
</tr>
<tr class="even">
<td align="left">ENSG00000159216</td>
<td align="left">RUNX1</td>
<td align="right">3</td>
<td align="left">Both</td>
<td align="right">0.6325</td>
<td align="right">4.192</td>
<td align="right">1.47e-05</td>
<td align="right">0.006</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000138696</td>
<td align="left">BMPR1B</td>
<td align="right">3</td>
<td align="left">Both</td>
<td align="right">-0.704</td>
<td align="right">3.476</td>
<td align="right">2.73e-05</td>
<td align="right">0.009</td>
</tr>
<tr class="even">
<td align="left">ENSG00000152766</td>
<td align="left">ANKRD22</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">0.5644</td>
<td align="right">5.298</td>
<td align="right">3.87e-05</td>
<td align="right">0.012</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000256229</td>
<td align="left">ZNF486</td>
<td align="right">1</td>
<td align="left">Both</td>
<td align="right">0.6752</td>
<td align="right">3.32</td>
<td align="right">7.83e-05</td>
<td align="right">0.021</td>
</tr>
<tr class="even">
<td align="left">ENSG00000149292</td>
<td align="left">TTC12</td>
<td align="right">2</td>
<td align="left">Both</td>
<td align="right">0.5598</td>
<td align="right">4.88</td>
<td align="right">8.64e-05</td>
<td align="right">0.022</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000049759</td>
<td align="left">NEDD4L</td>
<td align="right">2</td>
<td align="left">Both</td>
<td align="right">0.4862</td>
<td align="right">6.147</td>
<td align="right">9.13e-05</td>
<td align="right">0.022</td>
</tr>
<tr class="even">
<td align="left">ENSG00000047644</td>
<td align="left">WWC3</td>
<td align="right">2</td>
<td align="left">Both</td>
<td align="right">0.4455</td>
<td align="right">7.579</td>
<td align="right">1.56e-04</td>
<td align="right">0.035</td>
</tr>
</tbody>
</table>
<pre class="r"><code>rnaseq %&gt;%
  mutate(
    gata3PeakInProm = gene_id %in% (
      chipGR %&gt;%
        join_overlap_intersect(h3k27ac) %&gt;%
        subset(class == &quot;Promoter&quot;) %&gt;%
        join_overlap_intersect(geneRanges) %&gt;%
        mcols() %&gt;%
        as.data.frame() %&gt;% 
        pull(&quot;gene&quot;)
    ),
    DE = FDR &lt; 0.05
  ) %&gt;%
  group_by(DE, gata3PeakInProm) %&gt;%
  tally() %&gt;%
  pivot_wider(names_from = DE, values_from = n) %&gt;%
  with(
    cbind(`FALSE`, `TRUE`)
  ) %&gt;%
  fisher.test() %&gt;%
  pander(
    caption = &quot;Fisher&#39;s Exact Test providing moderate supprt for enrichment of GATA3 peaks within the promoters of DE genes, using an FDR of 0.05 to consider a gene as DE&quot;
  )</code></pre>
<table style="width:69%;">
<caption>Fisher’s Exact Test providing moderate supprt for enrichment of GATA3 peaks within the promoters of DE genes, using an FDR of 0.05 to consider a gene as DE</caption>
<colgroup>
<col width="16%" />
<col width="34%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">P value</th>
<th align="center">Alternative hypothesis</th>
<th align="center">odds ratio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">0.03525 *</td>
<td align="center">two.sided</td>
<td align="center">1.784</td>
</tr>
</tbody>
</table>
<div id="zbtb16" class="section level4">
<h4>ZBTB16</h4>
<p>The gene ZBTB16 has a gain of GATA3 in it’s promoter under DHT treatment, as well as the gain of an H3K27Ac peak under DHT treatment. It is also one of the most significantly up regulated genes in response to DHT. Whilst an interaction between the ZBTB16 promoter and a downstream region was detected, this region did not interact with any other GATA peaks and appeared inconclusive.</p>
<p>However, it is clear that there are:</p>
<ul>
<li>Synchronous gains of both GATA3 and AR in the upstream enhancer, along with increased H3K27Ac enhancer marks for two of these three enhancers</li>
<li>Displaced gains of AR and GATA3 in the upstream promoter, but with no increase in H3K27Ac signal</li>
<li>Increased GATA3 without AR in the direct promoter, along with increased H3K27Ac signal in the promoter</li>
</ul>
<pre class="r"><code>gr &lt;- chipGR %&gt;%
  join_overlap_intersect(h3k27ac) %&gt;%
  subset(class == &quot;Promoter&quot;) %&gt;%
  join_overlap_intersect(geneRanges) %&gt;%
  subset(symbol == &quot;ZBTB16&quot;)
plotRegion(gr, zoom = 1000, gm = transModels, shift = 1e5)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/chip_integration.Rmd/plotZBTB16-1.png" alt="*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. THe rectangular box highlights the relevant GATA3 site*" width="960" />
<p class="caption">
<em>Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. THe rectangular box highlights the relevant GATA3 site</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotZBTB16-1">
Past versions of plotZBTB16-1.png
</button>
</p>
<div id="fig-plotZBTB16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/e120d5420b7952e0b78fa9f622d985b23d3b5857/docs/figure/chip_integration.Rmd/plotZBTB16-1.png" target="_blank">e120d54</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotRegion(gr, zoom = 300, gm = transModels, shift = -5e3)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/chip_integration.Rmd/plotZBTB16_zoom-1.png" alt="*Zoomed in version of the above plot.*" width="960" />
<p class="caption">
<em>Zoomed in version of the above plot.</em>
</p>
</div>
</div>
<div id="ranbp3l" class="section level4">
<h4>RANBP3L</h4>
<p>The most highly ranked DE gene with an AR-independent GATA3 gain under DHT was <em>RANBP3L</em>.</p>
<pre class="r"><code>gr &lt;- chipGR %&gt;%
  join_overlap_intersect(h3k27ac) %&gt;%
  subset(class == &quot;Promoter&quot;) %&gt;%
  join_overlap_intersect(geneRanges) %&gt;%
  subset(symbol == &quot;RANBP3L&quot;)
plotRegion(gr, zoom = 1000, gm = transModels)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/chip_integration.Rmd/plotRANBP3L-1.png" alt="*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. THe rectangular box highlights the relevant GATA3 site*" width="960" />
<p class="caption">
<em>Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. THe rectangular box highlights the relevant GATA3 site</em>
</p>
</div>
</div>
<div id="ehf" class="section level4">
<h4>EHF</h4>
<p>The second most highly ranked DE gene with an AR-independent GATA3 gain under DHT was <em>EHF</em>. There was a discrepancy with the peak assignment to group, with four peaks being considered as DHT only, but some were found in both.</p>
<pre class="r"><code>gr &lt;- chipGR %&gt;%
  join_overlap_intersect(h3k27ac) %&gt;%
  subset(class == &quot;Promoter&quot;) %&gt;%
  join_overlap_intersect(geneRanges) %&gt;%
  subset(symbol == &quot;EHF&quot;) %&gt;%
  .[1:2] %&gt;%
  range()
plotRegion(gr, zoom = 20, gm = transModels)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/chip_integration.Rmd/plotEHF-1.png" alt="*Zoomed view of EHF showing the two peaks assigned as AR-independent DHT-responsive. The first was clearly detected in GATA3 (Vehicle) as well as GATA3 + DHT, whilst the second one was corectly annotated.*" width="960" />
<p class="caption">
<em>Zoomed view of EHF showing the two peaks assigned as AR-independent DHT-responsive. The first was clearly detected in GATA3 (Vehicle) as well as GATA3 + DHT, whilst the second one was corectly annotated.</em>
</p>
</div>
<pre class="r"><code>plotRegion(gr, zoom = 420, gm = transModels, shift = 8e4)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/chip_integration.Rmd/plotEHFExpanded-1.png" alt="*Expanded view of the above showing HiC contact locations*" width="960" />
<p class="caption">
<em>Expanded view of the above showing HiC contact locations</em>
</p>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] goseq_1.42.0                geneLenDataBase_1.26.0     
 [3] BiasedUrn_1.07              plyranges_1.10.0           
 [5] magrittr_1.5                GenomicInteractions_1.24.0 
 [7] InteractionSet_1.18.0       SummarizedExperiment_1.20.0
 [9] Biobase_2.50.0              MatrixGenerics_1.2.0       
[11] matrixStats_0.57.0          scales_1.1.1               
[13] pander_0.6.3                Rsamtools_2.6.0            
[15] Biostrings_2.58.0           XVector_0.30.0             
[17] glue_1.4.2                  Gviz_1.34.0                
[19] forcats_0.5.0               stringr_1.4.0              
[21] dplyr_1.0.2                 purrr_0.3.4                
[23] readr_1.4.0                 tidyr_1.1.2                
[25] tibble_3.0.4                ggplot2_3.3.2              
[27] tidyverse_1.3.0             rtracklayer_1.50.0         
[29] GenomicRanges_1.42.0        GenomeInfoDb_1.26.0        
[31] IRanges_2.24.0              S4Vectors_0.28.0           
[33] BiocGenerics_0.36.0         workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1             backports_1.2.0          Hmisc_4.4-1             
  [4] BiocFileCache_1.14.0     igraph_1.2.6             lazyeval_0.2.2          
  [7] splines_4.0.3            BiocParallel_1.24.0      digest_0.6.27           
 [10] ensembldb_2.14.0         htmltools_0.5.0          GO.db_3.12.1            
 [13] fansi_0.4.1              checkmate_2.0.0          memoise_1.1.0           
 [16] BSgenome_1.58.0          cluster_2.1.0            modelr_0.1.8            
 [19] askpass_1.1              prettyunits_1.1.1        jpeg_0.1-8.1            
 [22] colorspace_1.4-1         blob_1.2.1               rvest_0.3.6             
 [25] rappdirs_0.3.1           haven_2.3.1              xfun_0.19               
 [28] crayon_1.3.4             RCurl_1.98-1.2           jsonlite_1.7.1          
 [31] survival_3.2-7           VariantAnnotation_1.36.0 gtable_0.3.0            
 [34] zlibbioc_1.36.0          DelayedArray_0.16.0      DBI_1.1.0               
 [37] Rcpp_1.0.5               progress_1.2.2           htmlTable_2.1.0         
 [40] foreign_0.8-79           bit_4.0.4                Formula_1.2-4           
 [43] htmlwidgets_1.5.2        httr_1.4.2               RColorBrewer_1.1-2      
 [46] ellipsis_0.3.1           pkgconfig_2.0.3          XML_3.99-0.5            
 [49] nnet_7.3-14              dbplyr_2.0.0             here_0.1                
 [52] tidyselect_1.1.0         rlang_0.4.8              later_1.1.0.1           
 [55] AnnotationDbi_1.52.0     munsell_0.5.0            cellranger_1.1.0        
 [58] tools_4.0.3              cli_2.1.0                generics_0.1.0          
 [61] RSQLite_2.2.1            broom_0.7.2              evaluate_0.14           
 [64] yaml_2.2.1               knitr_1.30               bit64_4.0.5             
 [67] fs_1.5.0                 AnnotationFilter_1.14.0  nlme_3.1-149            
 [70] whisker_0.4              xml2_1.3.2               biomaRt_2.46.0          
 [73] compiler_4.0.3           rstudioapi_0.11          curl_4.3                
 [76] png_0.1-7                reprex_0.3.0             stringi_1.5.3           
 [79] highr_0.8                GenomicFeatures_1.42.0   lattice_0.20-41         
 [82] ProtGenerics_1.22.0      Matrix_1.2-18            vctrs_0.3.4             
 [85] pillar_1.4.6             lifecycle_0.2.0          BiocManager_1.30.10     
 [88] data.table_1.13.2        bitops_1.0-6             httpuv_1.5.4            
 [91] R6_2.5.0                 latticeExtra_0.6-29      promises_1.1.1          
 [94] gridExtra_2.3            dichromat_2.0-0          assertthat_0.2.1        
 [97] openssl_1.4.3            rprojroot_1.3-2          withr_2.3.0             
[100] GenomicAlignments_1.26.0 GenomeInfoDbData_1.2.4   mgcv_1.8-33             
[103] hms_0.5.3                rpart_4.1-15             rmarkdown_2.5           
[106] git2r_0.27.1             biovizBase_1.38.0        lubridate_1.7.9         
[109] base64enc_0.1-3         </code></pre>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
