<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stephen Pederson Dame Roma Mitchell Cancer Research Laboratories Adelaide Medical School University of Adelaide" />


<title>Dectection of Super-Enhancers Using ROSE</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ENCODE_T47D_HiC_ENCSR549MGQ</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="description.html">Workflow Description</a>
</li>
<li>
  <a href="annotation_setup.html">Annotations</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="interactions_setup.html">Setup</a>
    </li>
    <li>
      <a href="interactions_40kb.html">40kb Interactions</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Super Enhancers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="rose.html">ROSE</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Dectection of Super-Enhancers Using ROSE</h1>
<h4 class="author">Stephen Pederson<br>Dame Roma Mitchell Cancer Research Laboratories<br>Adelaide Medical School<br>University of Adelaide</h4>
<h4 class="date">25 November, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-11-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>ENCODE_T47D_HiC_ENCSR549MGQ/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200918code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200918)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200918code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200918)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLENCODET47DHiCENCSR549MGQtree8e0b97c9a21d0cb08e3b4b8b9dbe3d17a3ea90datargetblank8e0b97ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/tree/8e0b97c9a21d0cb08e3b4b8b9dbe3d17a3ea90da" target="_blank">8e0b97c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLENCODET47DHiCENCSR549MGQtree8e0b97c9a21d0cb08e3b4b8b9dbe3d17a3ea90datargetblank8e0b97ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/tree/8e0b97c9a21d0cb08e3b4b8b9dbe3d17a3ea90da" target="_blank">8e0b97c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .snakemake/
    Ignored:    data/external/H3K27AC/bam/
    Ignored:    data/external/H3K27AC/bigwig/
    Ignored:    data/external/gencode.v33lift37.annotation.gtf.gz
    Ignored:    output/MaxHiC/

Unstaged changes:
    Modified:   analysis/chip_integration.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/rose.Rmd</code>) and HTML (<code>docs/rose.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/8e0b97c9a21d0cb08e3b4b8b9dbe3d17a3ea90da/analysis/rose.Rmd" target="_blank">8e0b97c</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-25
</td>
<td>
Changed export to all enhancers
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/bae52d2e0b44fbddaf774ba3868b7a991bd0a3b8/docs/rose.html" target="_blank">bae52d2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/de66e545baaad535e982e2608e9b9b28bda26032/analysis/rose.Rmd" target="_blank">de66e54</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Removed stitched SEs that overlapped a promoter
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/9ce4414c70630e544024bedfbcba90d73c704f92/docs/rose.html" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/2bbc2ccf05f809868ae93e49256b8b9b7a2f488e/analysis/rose.Rmd" target="_blank">2bbc2cc</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Finished checking SEs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/0c4e66a99028ddc7edd459d2c36b16ddaecf05b9/docs/rose.html" target="_blank">0c4e66a</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/55adffb8389021e3c5d5b20aaf573b2db32a414f/analysis/rose.Rmd" target="_blank">55adffb</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Resized figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/2491240e00ca86842fb5105c26622bbe5ba178ee/docs/rose.html" target="_blank">2491240</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/ac9288ef60ecb60ae79312f3406af27bb36c26e0/analysis/rose.Rmd" target="_blank">ac9288e</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Finished ROSE analysis. Workflowr Error during build, so will reboot later.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/ac9288ef60ecb60ae79312f3406af27bb36c26e0/docs/rose.html" target="_blank">ac9288e</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
<td>
Finished ROSE analysis. Workflowr Error during build, so will reboot later.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(rtracklayer)
library(tidyverse)
library(Gviz)
library(glue)
library(Rsamtools)
library(pander)
library(scales)
library(GenomicInteractions)
library(InteractionSet)
library(magrittr)</code></pre>
<pre class="r"><code>theme_set(theme_bw())
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)</code></pre>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This is a manual re-implementation of the ROSE algorithm and relies on ChIP peaks from H3K21AC data being previously called. These are assumed to be candidate regions for both promoters and enhancers.</p>
<div id="required-data" class="section level2">
<h2>Required Data</h2>
<p>Before running the ROSE algorithm, the following files must be sourced</p>
<ol style="list-style-type: decimal">
<li>Consensus Peaks for each treatment as <code>.bed</code> files</li>
<li>Merged <code>.bam</code> files for all replicates within each treatment group and the control sample</li>
<li>BigWig files for each replicate, purely for the purposes of visualisation</li>
</ol>
</div>
<div id="steps-of-the-rose-algorithm" class="section level2">
<h2>Steps of the ROSE Algorithm</h2>
<p>The steps are as follows:</p>
<ol style="list-style-type: decimal">
<li>Remove any regions which overlap a TSS and set aside as promoters
<ul>
<li><em>Promoters were restricted to genes detected as expressed in the associated RNA-Seq dataset</em></li>
<li>TSS regions are extended by 1.5kb upstream and 500bp downstream to give a broader 2kb TSS/promoter region, however the size of the peaks effectively defines the promoter</li>
</ul></li>
<li>Join any remaining candidate enhancers within 12.5kb
<ul>
<li>Remove any candidate regions shorter than 1kb as these will not make the cut</li>
</ul></li>
<li>Count reads aligned within each region to get the number of reads
<ul>
<li>Alignments were pre-filtered such that only reads with a <code>MAPQ</code> score &gt; 10 were retained</li>
<li>Strandedness of aligned reads was not taken into account</li>
</ul></li>
<li>Counts are transformed into Reads Per Million (<code>rpm</code>) by dividing by the total number of reads aligned within each bam file, divided by 10<sup>-6</sup></li>
<li><code>rpm</code> values within each region are divided by the region width to give <code>rpm/bp</code></li>
<li><code>rpm/bp</code> values are then background corrected by subtracting the values observed in the control sample
<ul>
<li>Any <code>rpm/bp</code> &lt; 0 are automatically set to 0 when calculating the cutoff</li>
</ul></li>
<li>Candidate enhancers are then ranked and the point found for which the slope of the tangent line drops below 1</li>
</ol>
<p>Unlike other implementations of ROSE, standard Bioconductor data structures will be used.</p>
<p>Additionally, the functions <code>calculate_cutoff()</code> and <code>numPts_below_line()</code> were taken directly from the main <a href="https://github.com/rakarnik/ROSE/blob/master/ROSE_callSuper.R">ROSE github software repository</a></p>
</div>
</div>
<div id="data-setup" class="section level1">
<h1>Data Setup</h1>
<div id="gene-and-transcript-models" class="section level2">
<h2>Gene and Transcript Models</h2>
<pre class="r"><code>rnaseq &lt;- here::here(&quot;data/external/RNASeq/t47d_DHT_StrippedSerum_RNASeq_topTable.tsv&quot;) %&gt;%
  read_tsv()</code></pre>
<pre class="r"><code>transModels &lt;- here::here(&quot;output/transModels.rds&quot;) %&gt;%
  read_rds()
tss &lt;- transModels %&gt;%
  subset(gene %in% rnaseq$gene_id) %&gt;%
  split(f = .$transcript) %&gt;%
  as(&quot;GRangesList&quot;) %&gt;%
  range() %&gt;%
  unlist() %&gt;%
  resize(width = 1, fix = &quot;start&quot;) %&gt;%
  promoters(downstream = 500, upstream = 1500) %&gt;%
  sort()</code></pre>
<pre class="r"><code>geneModels &lt;- here::here(&quot;output/geneModels.rds&quot;) %&gt;%
  read_rds()
geneRanges &lt;- geneModels %&gt;% 
  split(f = .$gene) %&gt;% 
  range() %&gt;% 
  unlist() %&gt;%
  sort()
geneRanges$gene &lt;- names(geneRanges)
geneRanges$symbol &lt;- geneModels$symbol[match(geneRanges$gene, geneModels$gene)]
sq &lt;- seqinfo(geneModels)</code></pre>
<p>Gene and transcript models were loaded for plotting and data integration. These were defined in the <a href="annotation_setup.html">Annotation Setup</a> step of the workflow. Gene models make for simpler plotting, whilst transcript models allow for more accurate identification of each TSS.</p>
<p>For this analysis, only the TSS of genes detected as expressed in the RNA-Seq dataset (Stripped Serum) were considered as viable TSS &amp; promoters. Any peaks which overlapped undetectable gene promoters were considered more likely to be functioning as enhancers.</p>
</div>
<div id="consensus-peaks" class="section level2">
<h2>Consensus Peaks</h2>
<pre class="r"><code>chipGR &lt;- here::here(&quot;data/external/BED&quot;) %&gt;%
  list.files(pattern = &quot;H3K27&quot;, full.names = TRUE) %&gt;%
  sapply(import.bed, seqinfo = sq) %&gt;%
  lapply(sort) %&gt;%
  setNames(
    str_replace(names(.), &quot;.+(E2.*)_peaks.bed&quot;, &quot;\\1&quot;)
  ) %&gt;%
  as(&quot;GRangesList&quot;) </code></pre>
<pre class="r"><code>consPeaks &lt;- chipGR %&gt;%
  as(&quot;GRangesList&quot;) %&gt;%
  unlist() %&gt;%
  GenomicRanges::reduce() </code></pre>
<p>Consensus peaks were identified for each treatment group (E2 and E2+DHT), however for the purposes of super-enhancer detection, <em>these were merged to ensure the same candidate super-enhancer was being investigated in each cell type</em>. Subtle shifts in ChIP peaks boundaries should remain relatively inconsequential for the relatively extreme criteria required for classification as a super-enhancer.</p>
<p>Whilst 44,044 and 45,654 consensus peaks were identified for E2 and E2+DHT respectively, when combining a total of 46,946 regions were defined as the merged consensus peaks of the separate sets of consensus peaks.</p>
</div>
<div id="promoters-and-enhancers" class="section level2">
<h2>Promoters and Enhancers</h2>
<pre class="r"><code>ol &lt;- findOverlaps(tss, consPeaks, ignore.strand = TRUE)
promGR &lt;- consPeaks[unique(subjectHits(ol))]
enhGR &lt;- consPeaks[-unique(subjectHits(ol))]
stitchGR &lt;- GenomicRanges::reduce(enhGR, min.gapwidth = 12500) %&gt;%
  subset(width &gt; 1e3)</code></pre>
<p>After comparison with promoters (defined as TSS <span class="math inline">\(\pm\)</span> 2500bp), 14,346 regions were classified as candidate promoters, whilst 32,600 were retained as candidate enhancers. After stitching together peaks within 12.5kb of each other and removing peaks shorter than 1kb, the final set of 10,759 candidate regions was formed. Notably, these regions were independent of the initial treatment group, but were identification of super-enhancers will be performed separately in each treatment group.</p>
<pre class="r"><code>cutPromoters &lt;- function(i, x = stitchGR, y = enhGR, p = promGR){
  
  ## To correct these ranges, we should first &#39;snip&#39; the promoter peak from the 
  ## merged range, but just putting a hole in the range
  ## The first offender is stitchGR[12]
  x &lt;- x[i]
  ## This merged 3 enhancers with the promoter being between 2 &amp; 3
  y &lt;- subsetByOverlaps(y, x)
  ## The promoter peak is
  p &lt;- subsetByOverlaps(p, x) %&gt;% range()
  ## Given we&#39;ll always have a range either side, we can just cut these
  xl &lt;- range(subset(y, end &lt; start(p)))
  xr &lt;- range(subset(y, start &gt; end(p)))
  ## Now we return the new ranges
  c(xl, xr)
}</code></pre>
<pre class="r"><code>ol &lt;- findOverlaps(stitchGR, promGR) %&gt;%
  queryHits() %&gt;%
  unique()</code></pre>
<p>An issue produced by this method, is although peaks overlapping promoters were excluded, peaks straddling either side of the peaks directly overlapping a promoter may still be joined. These were identified and any merged peaks with this characteristic were split either side of the overlapping peak. A total of 535 stitched regions were found to overlap a promoter and required ‘unstitching’ back into two separate regions either side of the promoter.</p>
<pre class="r"><code>nc &lt;- detectCores() - 1
unstitched &lt;- ol %&gt;%
  mclapply(cutPromoters, mc.cores = nc) %&gt;%
  as(&quot;GRangesList&quot;) %&gt;%
  unlist()
stitchGR &lt;- c(
  stitchGR[-ol],
  unstitched
) %&gt;%
  sort() %&gt;%
  subset(width &gt; 1e3)</code></pre>
</div>
<div id="alignments" class="section level2">
<h2>Alignments</h2>
<pre class="r"><code>bfl &lt;- here::here(&quot;data/external/H3K27AC/bam&quot;) %&gt;%
  list.files(pattern= &quot;bam$&quot;, recursive = TRUE, full.names = TRUE) %&gt;%
  BamFileList()</code></pre>
<pre class="r"><code>treat &lt;- names(bfl) %&gt;%
  str_replace_all(&quot;.+(E2.*|Pooled)_(input|merged).+&quot;, &quot;\\1&quot;) %&gt;%
  setNames(names(bfl))</code></pre>
<pre class="r"><code>libSizes &lt;- bfl %&gt;%
  lapply(idxstatsBam) %&gt;%
  lapply(pull, &quot;mapped&quot;) %&gt;%
  vapply(sum, numeric(1)) %&gt;%
  divide_by(1e6)</code></pre>
<p>The <code>.bam</code> files were assigned to a <code>BamFileList</code> object for simple parsing of each range.</p>
<pre class="r"><code>pander(enframe(comma(1e6*libSizes), name = &quot;BamFile&quot;, value = &quot;Total Alignments&quot;), caption = &quot;Summary of alignments within each treatment group&quot;)</code></pre>
<table style="width:78%;">
<caption>Summary of alignments within each treatment group</caption>
<colgroup>
<col width="51%" />
<col width="26%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">BamFile</th>
<th align="center">Total Alignments</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">T47D_H3K27Ac_E2_DHT_merged.bam</td>
<td align="center">96,596,880</td>
</tr>
<tr class="even">
<td align="center">T47D_H3K27Ac_E2_merged.bam</td>
<td align="center">58,163,939</td>
</tr>
<tr class="odd">
<td align="center">T47D_Pooled_input_mapped_reads.bam</td>
<td align="center">37,464,498</td>
</tr>
</tbody>
</table>
</div>
<div id="function-definitions" class="section level2">
<h2>Function Definitions</h2>
<pre class="r"><code>calculate_cutoff &lt;- function(inputVector, drawPlot=TRUE,...){
  inputVector &lt;- sort(inputVector)
  inputVector[inputVector&lt;0]&lt;-0 #set those regions with more control than ranking equal to zero
  slope &lt;- (max(inputVector)-min(inputVector))/length(inputVector) #This is the slope of the line we want to slide. This is the diagonal.
  xPt &lt;- floor(optimize(numPts_below_line,lower=1,upper=length(inputVector),myVector= inputVector,slope=slope)$minimum) #Find the x-axis point where a line passing through that point has the minimum number of points below it. (ie. tangent)
  y_cutoff &lt;- inputVector[xPt] #The y-value at this x point. This is our cutoff.
  
  if(drawPlot){  #if TRUE, draw the plot
    plot(1:length(inputVector), inputVector,type=&quot;l&quot;,...)
    b &lt;- y_cutoff-(slope* xPt)
    abline(v= xPt,h= y_cutoff,lty=2,col=8)
    points(xPt,y_cutoff,pch=16,cex=0.9,col=2)
    abline(coef=c(b,slope),col=2)
    text(x = 0.25*length(inputVector), y = 0.9*max(inputVector), labels = paste(&quot;x=&quot;,xPt,&quot;\ny=&quot;,signif(y_cutoff,3),&quot;\nFold over Median=&quot;,signif(y_cutoff/median(inputVector),3),&quot;x\nFold over Mean=&quot;,signif(y_cutoff/mean(inputVector),3),&quot;x&quot;,sep=&quot;&quot;))
    axis(1,sum(inputVector==0),sum(inputVector==0),col.axis=&quot;pink&quot;,col=&quot;pink&quot;) #Number of regions with zero signal
  }
  return(list(absolute=y_cutoff,overMedian=y_cutoff/median(inputVector),overMean=y_cutoff/mean(inputVector)))
}</code></pre>
<pre class="r"><code>numPts_below_line &lt;- function(myVector,slope,x){
  yPt &lt;- myVector[x]
  b &lt;- yPt-(slope*x)
  xPts &lt;- 1:length(myVector)
  return(sum(myVector&lt;=(xPts*slope+b)))
}</code></pre>
<p>As stated earlier, the two functions <code>calculate_cutoff()</code> and <code>numPts_below_line()</code> were copied directly from the <a href="https://github.com/rakarnik/ROSE/blob/master/ROSE_callSuper.R">ROSE github software repository</a></p>
</div>
</div>
<div id="detection-of-super-enhancers" class="section level1">
<h1>Detection of Super-Enhancers</h1>
<div id="calculation-of-enrichment-within-regions" class="section level2">
<h2>Calculation of Enrichment Within Regions</h2>
<pre class="r"><code>rpmpbp &lt;- countBam(bfl, param = ScanBamParam(which = stitchGR)) %&gt;%
  mutate(
    treat = treat[file],
    rpmpbp = (records / libSizes[file]) / width
  ) %&gt;%
  pivot_wider(
    id_cols = c(&quot;space&quot;, &quot;start&quot;, &quot;end&quot;, &quot;width&quot;),
    names_from = treat,
    names_prefix = &quot;rpmpbp_&quot;,
    values_from = rpmpbp
  ) %&gt;%
  mutate(
    across(contains(&quot;rpm&quot;), function(x){x - .$rpmpbp_Pooled})
  ) %&gt;%
  dplyr::select(-contains(&quot;Pooled&quot;)) %&gt;%
  dplyr::filter(
    rpmpbp_E2_DHT &gt; 0,
    rpmpbp_E2 &gt; 0
  )</code></pre>
<pre class="r"><code>rpmpbp %&gt;%
  ggplot(aes(rpmpbp_E2_DHT, rpmpbp_E2)) + 
  geom_point() + 
  geom_smooth(se = FALSE) +
  geom_abline(slope = 1, linetype = 2, colour = &quot;grey&quot;) +
  labs(
    x = &quot;E2 + DHT (rpm/bp)&quot;,
    y = &quot;E2 (rpm/bp)&quot;
  ) +
  scale_x_sqrt() +
  scale_y_sqrt()</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/rose.Rmd/plotRpmpbp-1.png" alt="Comparison of H3K27AC signal within each candidate region. The curve of best fit is shown in blue, with y = x shown as the dashed, grey line. Axes are both square-root transformed." width="768" />
<p class="caption">
Comparison of H3K27AC signal within each candidate region. The curve of best fit is shown in blue, with y = x shown as the dashed, grey line. Axes are both square-root transformed.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotRpmpbp-1">
Past versions of plotRpmpbp-1.png
</button>
</p>
<div id="fig-plotRpmpbp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/bae52d2e0b44fbddaf774ba3868b7a991bd0a3b8/docs/figure/rose.Rmd/plotRpmpbp-1.png" target="_blank">bae52d2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/9ce4414c70630e544024bedfbcba90d73c704f92/docs/figure/rose.Rmd/plotRpmpbp-1.png" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/0c4e66a99028ddc7edd459d2c36b16ddaecf05b9/docs/figure/rose.Rmd/plotRpmpbp-1.png" target="_blank">0c4e66a</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/2491240e00ca86842fb5105c26622bbe5ba178ee/docs/figure/rose.Rmd/plotRpmpbp-1.png" target="_blank">2491240</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Reads per million / base pair (<code>rpm/bp</code>) were calculated as described above for each candidate region, with any receiving a background corrected value &lt; 0 removed. Values were then visually compared across treatments. In the lower regions, signal strongly tracked the line <code>y = x</code>, whilst for the regions with the strongest signal, a trend of higher signal was observed in the merged E2 alignments. This may represent bias in peak definitions (e.g. wider peaks predominately defined by one treatment group), the impact of the larger library size in the E2+DHT sample, or may even be genuine biology.</p>
</div>
<div id="detection-of-super-enhancers-1" class="section level2">
<h2>Detection of Super-Enhancers</h2>
<pre class="r"><code>par(mfrow = c(1, 2))
cuts &lt;- c()
cuts$E2 &lt;- calculate_cutoff(rpmpbp$rpmpbp_E2, main = &quot;E2&quot;, ylab = &quot;H3K27AC Signal (rpm/bp)&quot;, xlab = &quot;Rank&quot;) %&gt;% 
  as_tibble() %&gt;% 
  mutate(treat = &quot;E2&quot;)
cuts$E2_DHT &lt;- calculate_cutoff(rpmpbp$rpmpbp_E2_DHT, main = &quot;E2 + DHT&quot;, ylab = &quot;H3K27AC Signal (rpm/bp)&quot;, xlab = &quot;Rank&quot;) %&gt;% 
  as_tibble() %&gt;% 
  mutate(treat = &quot;E2_DHT&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/rose.Rmd/plotRose-1.png" alt="Super-Enhancer Detection Using ROSE" width="960" />
<p class="caption">
Super-Enhancer Detection Using ROSE
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotRose-1">
Past versions of plotRose-1.png
</button>
</p>
<div id="fig-plotRose-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/bae52d2e0b44fbddaf774ba3868b7a991bd0a3b8/docs/figure/rose.Rmd/plotRose-1.png" target="_blank">bae52d2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/9ce4414c70630e544024bedfbcba90d73c704f92/docs/figure/rose.Rmd/plotRose-1.png" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/0c4e66a99028ddc7edd459d2c36b16ddaecf05b9/docs/figure/rose.Rmd/plotRose-1.png" target="_blank">0c4e66a</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/2491240e00ca86842fb5105c26622bbe5ba178ee/docs/figure/rose.Rmd/plotRose-1.png" target="_blank">2491240</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>par(mfrow = c(1,1))
cuts %&lt;&gt;% bind_rows()</code></pre>
<pre class="r"><code>rpmpbp %&lt;&gt;%
  mutate(
    class = case_when(
      rpmpbp_E2_DHT &gt; dplyr::filter(cuts, treat == &quot;E2_DHT&quot;)$absolute &amp; 
        rpmpbp_E2 &gt; dplyr::filter(cuts, treat == &quot;E2&quot;)$absolute ~ &quot;Both&quot;,
      rpmpbp_E2_DHT &gt; dplyr::filter(cuts, treat == &quot;E2_DHT&quot;)$absolute ~ &quot;E2 + DHT Only&quot;,
      rpmpbp_E2 &gt; dplyr::filter(cuts, treat == &quot;E2&quot;)$absolute ~ &quot;E2 Only&quot;,
      TRUE ~ &quot;Not SE&quot;
    )
  ) </code></pre>
<pre class="r"><code>rpmpbp %&gt;%
  group_by(class) %&gt;%
  summarise(
    n = n(),
    Smallest = min(width),
    `median width` = median(width),
    Largest = max(width)
  ) %&gt;%
  rename_all(str_to_title) %&gt;%
  pander(
    caption = &quot;Summary of Super-Enhancers detected by treatment. The median width for each class is also given&quot;
  )</code></pre>
<table style="width:83%;">
<caption>Summary of Super-Enhancers detected by treatment. The median width for each class is also given</caption>
<colgroup>
<col width="22%" />
<col width="11%" />
<col width="15%" />
<col width="20%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Class</th>
<th align="center">N</th>
<th align="center">Smallest</th>
<th align="center">Median Width</th>
<th align="center">Largest</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Both</td>
<td align="center">958</td>
<td align="center">1,033</td>
<td align="center">3,172</td>
<td align="center">131,297</td>
</tr>
<tr class="even">
<td align="center">E2 + DHT Only</td>
<td align="center">213</td>
<td align="center">1,069</td>
<td align="center">2,704</td>
<td align="center">40,359</td>
</tr>
<tr class="odd">
<td align="center">E2 Only</td>
<td align="center">29</td>
<td align="center">1,442</td>
<td align="center">5,375</td>
<td align="center">51,239</td>
</tr>
<tr class="even">
<td align="center">Not SE</td>
<td align="center">9,678</td>
<td align="center">1,001</td>
<td align="center">2,762</td>
<td align="center">207,654</td>
</tr>
</tbody>
</table>
<p>Initial analysis revealed that many more super-enhancers were identified in the DHT treated samples, which is counter-intuitive to the observations made with regard to raw signal. It was also noted that the median length of the super enhancers identified in both treatment groups was much longer than the tissue-specific SEs. A comparison of H3K27Ac signal for each region was then performed.</p>
<pre class="r"><code>rpmpbp %&gt;%
  ggplot(aes(rpmpbp_E2_DHT, rpmpbp_E2)) +
  geom_point(aes(colour = class)) +
  geom_smooth(se = FALSE) +
  geom_abline(slope = 1, linetype = 2) +
  geom_vline(xintercept = dplyr::filter(cuts, treat == &quot;E2_DHT&quot;)$absolute, colour = &quot;grey&quot;, linetype = 2) +
  geom_hline(yintercept = dplyr::filter(cuts, treat == &quot;E2&quot;)$absolute, colour = &quot;grey&quot;, linetype = 2) +
  scale_y_sqrt() +
  scale_x_sqrt() +
  scale_colour_manual(
    values = c(rgb(0.1, 0.8, 0.1), rgb(0.8, 0.2,0.2), rgb(0.2, 0.2, 0.8), rgb(0.7, 0.7,0.7,0.5))
  ) +
  coord_cartesian(
    xlim = c(0.001, 0.03),
    ylim = c(0.001, 0.03)
  ) +
  labs(
    x = &quot;E2 + DHT (rpm/bp)&quot;,
    y = &quot;E2 (rpm/bp)&quot;,
    colour = &quot;SE Detection&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/rose.Rmd/compareSE-1.png" alt="Comparison of H3K27AC signal in the two treatment groups for each type of super-enhancer. ROSE thresholds for inclusion as a super enhancers are shown as grey dashed lines." width="768" />
<p class="caption">
Comparison of H3K27AC signal in the two treatment groups for each type of super-enhancer. ROSE thresholds for inclusion as a super enhancers are shown as grey dashed lines.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compareSE-1">
Past versions of compareSE-1.png
</button>
</p>
<div id="fig-compareSE-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/bae52d2e0b44fbddaf774ba3868b7a991bd0a3b8/docs/figure/rose.Rmd/compareSE-1.png" target="_blank">bae52d2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/9ce4414c70630e544024bedfbcba90d73c704f92/docs/figure/rose.Rmd/compareSE-1.png" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/0c4e66a99028ddc7edd459d2c36b16ddaecf05b9/docs/figure/rose.Rmd/compareSE-1.png" target="_blank">0c4e66a</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/2491240e00ca86842fb5105c26622bbe5ba178ee/docs/figure/rose.Rmd/compareSE-1.png" target="_blank">2491240</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>After checking the plot above, it appears that an increase in SE activity can be partially attributed to DHT treatment. A series of putative super enhancers detected in the E2 + DHT treatment group fall non-trivially below the cutoff value for the E2 only treatment group, whilst the converse is not observed. Those classified as E2 only clearly appear artefactual, and whilst many specific to the E2 + DHT group also appear so, a select group of super enhancers do not.</p>
<p>The detected set of super-enhancers were exported, along with the original set of peaks divided into promoters and enhancers.</p>
<pre class="r"><code>seGR &lt;- rpmpbp %&gt;%
  dplyr::filter(class != &quot;Not SE&quot;) %&gt;%
  makeGRangesFromDataFrame(
    seqnames.field = &quot;space&quot;,
    keep.extra.columns = TRUE,
    seqinfo = sq
  )
export.gff(seGR, &quot;output/SE.gtf&quot;)
export.bed(enhGR, &quot;output/enhancers.bed&quot;)
export.bed(promGR, &quot;output/promoters.bed&quot;)
ol &lt;- findOverlaps(enhGR, seGR)
enhGR &lt;- enhGR[-unique(queryHits(ol))]</code></pre>
<p>A comparison of region widths was then performed and it was noted that some regions annotated as promoters were extremely long. This may be the result of some genes containing multiple TSS, but also the original peak calling may have merged multiple peaks into a larger peak.</p>
<pre class="r"><code>list(
  tibble(width = width(promGR), region = &quot;Promoter&quot;),
   tibble(width = width(enhGR), region = &quot;Enhancer&quot;),
   tibble(width = width(seGR), region = &quot;Super Enhancer&quot;)
) %&gt;%
  bind_rows() %&gt;%
  mutate(region = fct_inorder(region)) %&gt;%
  ggplot(aes(region, width)) +
  geom_boxplot() +
  labs(x = &quot;Region&quot;, y = &quot;Region Width (bp)&quot;) +
  scale_y_log10(labels = scales::comma)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/rose.Rmd/compareWidths-1.png" alt="*Comparison of region widths for promoters, enhancers and super-enhancers after running the above processes.*" width="768" />
<p class="caption">
<em>Comparison of region widths for promoters, enhancers and super-enhancers after running the above processes.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compareWidths-1">
Past versions of compareWidths-1.png
</button>
</p>
<div id="fig-compareWidths-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/bae52d2e0b44fbddaf774ba3868b7a991bd0a3b8/docs/figure/rose.Rmd/compareWidths-1.png" target="_blank">bae52d2</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/9ce4414c70630e544024bedfbcba90d73c704f92/docs/figure/rose.Rmd/compareWidths-1.png" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="inspection-of-results" class="section level1 tabset">
<h1>Inspection of Results</h1>
<p>After identification of SEs, regions considered as SE in any condition were exported as a distinct set of regions, whilst enhancers not considered as an SE were exported as a distinct set of enhancers. This gave three distinct types of binding regions: 1) Promoters, 2) Enhancers and 3) Super Enhancers. Individual ChIP Peaks and BigWig files (i.e. scaled coverage) were inspected at key sites, incorporating HiC information and manually checking RNA-Seq expression.</p>
<ul>
<li>Significant interactions detected in the HiC data using bins of 20kb, 40kb and 100kb were combined into a single set of interactions</li>
<li>The ChromHMM model for HMEC cells was also prepared as a track for plotting</li>
<li>In all plots below, the RNA-Seq data was combined with the H3K27AC data to give four basic classifications for peaks and known exons.
<ol style="list-style-type: decimal">
<li>Active Promoter</li>
<li>Transcription</li>
<li>Enhancer</li>
<li>Super Enhancer</li>
</ol></li>
<li>Colours for the above were identical to the ChromHMM model, with the exception of <em>colouring super-enhancers blue.</em></li>
</ul>
<pre class="r"><code>bw &lt;- here::here(&quot;data/external/H3K27AC/bigwig&quot;) %&gt;%
  list.files(pattern = &quot;_[1-3].bigwig&quot;, full.names = TRUE) %&gt;%
  BigWigFileList() %&gt;%
  setNames(
    str_replace_all(path(.), &quot;.+T47D_(.+).bigwig&quot;, &quot;\\1&quot;)
  )
bwCols &lt;- case_when(
  grepl(&quot;DHT&quot;, names(bw)) ~&quot;red&quot;, 
  grepl(&quot;Pooled&quot;, names(bw)) ~ &quot;grey&quot;,
  grepl(&quot;E2_[1-3]&quot;, names(bw)) ~ &quot;blue&quot;
) %&gt;%
  setNames(names(bw))</code></pre>
<pre class="r"><code>stateMap &lt;- read_rds(here::here(&quot;output/stateMap.rds&quot;))
chromCols &lt;- setNames(stateMap$itemRgb, stateMap$name)
hmecHMM &lt;- import.bed(&quot;data/external/ChromHMM/E119_25_imputed12marks_dense.bed.gz&quot;, seqinfo = sq) %&gt;%
  sort()</code></pre>
<pre class="r"><code>gi100 &lt;- here::here(&quot;output/gi_100000.rds&quot;) %&gt;% read_rds()
gi40 &lt;- here::here(&quot;output/gi_40000.rds&quot;) %&gt;% read_rds()
gi20 &lt;- here::here(&quot;output/gi_20000.rds&quot;) %&gt;% read_rds()
gi10 &lt;- here::here(&quot;output/gi_10000.rds&quot;) %&gt;% read_rds()
gi &lt;- c(gi10, gi20, gi40, gi100)
gi$distance &lt;- pairdist(gi)
gi$logObsExp &lt;- log2(gi$counts / gi$exp_interactions)</code></pre>
<pre class="r"><code>gen &lt;- &quot;hg19&quot;
ax &lt;- GenomeAxisTrack()</code></pre>
<pre class="r"><code>plotRegion &lt;- function(gr, zoom = 1, hic = gi, hmm = hmecHMM, prom = promGR, enh = enhGR, se = seGR, gm = geneModels, .bw = bw, .bwCols = bwCols, .ax = ax, rna = rnaseq, maxIntDist = 1e6){
  
  stopifnot(is(gr, &quot;GRanges&quot;) | length(gr) &gt; 0)
  
  ## Form the Ideogram track
  chr &lt;- unique(as.character(seqnames(gr)))
  ideo &lt;- IdeogramTrack(genome = gen, chromosome = chr)
  
  ## The HiC Track
  goiInteractions &lt;- hic %&gt;%
    subsetByOverlaps(gr, ignore.strand = TRUE) %&gt;%
    # subset(distance &lt; width(gr)*zoom) %&gt;%
    subset(distance &lt; maxIntDist) %&gt;%
    as(&quot;GenomicInteractions&quot;)
  hicTR &lt;- goiInteractions %&gt;%
    InteractionTrack(
      chromosome = chr,
      name = &quot;HiC&quot;
    )
  
  ## Expand the range  
  gr &lt;- resize(gr, width = zoom*width(gr), fix = &quot;center&quot;)
  
  ## The ChromHMM track
  hmmTR &lt;- hmm %&gt;%
    subsetByOverlaps(gr, ignore.strand = TRUE) %&gt;%
    AnnotationTrack(
      name = &quot;HMEC&quot;,
      id = .$name,
      stacking = &quot;dense&quot;,
      feature = .$name,
      group = .$name,
      col = &quot;transparent&quot;,
      `1_TssA` = &quot;#FF0000&quot;, `10_TxEnh5&#39;` = &quot;#C2E105&quot;, `11_TxEnh3&#39;` = &quot;#C2E105&quot;,
      `12_TxEnhW` = &quot;#C2E105&quot;, `13_EnhA1` = &quot;#FFC34D&quot;, `14_EnhA2` = &quot;#FFC34D&quot;,
      `15_EnhAF` = &quot;#FFC34D&quot;, `16_EnhW1` = &quot;#FFFF00&quot;, `17_EnhW2` = &quot;#FFFF00&quot;,
      `18_EnhAc` = &quot;#FFFF00&quot;, `19_DNase` = &quot;#FFFF66&quot;, `2_PromU` = &quot;#FF4500&quot;,
      `20_ZNF/Rpts` = &quot;#66CDAA&quot;, `21_Het` = &quot;#8A91D0&quot;, `22_PromP` = &quot;#E6B8B7&quot;,
      `23_PromBiv` = &quot;#7030A0&quot;, `24_ReprPC` = &quot;#808080&quot;, `25_Quies` = &quot;#FFFFFF&quot;,
      `3_PromD1` = &quot;#FF4500&quot;, `4_PromD2` = &quot;#FF4500&quot;, `5_Tx5&#39;` = &quot;#008000&quot;,
      `6_Tx` = &quot;#008000&quot;, `7_Tx3&#39;` = &quot;#008000&quot;, `8_TxWk` = &quot;#009600&quot;,
      `9_TxReg` = &quot;#C2E105&quot;
    )
  
  # Setup the promoters, enhancers and super-enhancers like a 3-state ChromHMM track.
  # Maybe include the detected genes as &#39;Transcription&#39;
  prom &lt;- granges(prom)
  prom$name &lt;- &quot;Promoter&quot;
  enh &lt;- granges(enh)
  enh$name &lt;- &quot;Enhancer&quot;
  se &lt;- granges(se)
  se$name &lt;- &quot;Super-Enhancer&quot;
  exons &lt;- subset(gm, gene %in% rna$gene_id)
  exons &lt;- granges(exons)
  exons$name &lt;- &quot;Transcription&quot;
  h3K27ac &lt;- GRangesList(prom, enh, se, exons) %&gt;%
    unlist() %&gt;%
    sort()
  strand(h3K27ac) &lt;- &quot;*&quot;
  noFeature &lt;- gaps(h3K27ac)
  noFeature &lt;- subset(noFeature, strand == &quot;*&quot;)
  noFeature$name &lt;- &quot;None&quot;
  h3K27ac &lt;- GRangesList(
    h3K27ac,
    noFeature
  ) %&gt;%
    unlist() %&gt;%
    sort()
  h3K27acTR &lt;- h3K27ac %&gt;%
    subsetByOverlaps(gr, ignore.strand = TRUE) %&gt;%
    AnnotationTrack(
      name = &quot;H3K27+RNA&quot;,
      stacking = &quot;dense&quot;,
      id = .$name,
      feature = .$name,
      group = .$name,
      col = &quot;transparent&quot;,
      fontsize = 12,
      size = 1.5,
      `Promoter` = &quot;#FF4500&quot;,
      `Enhancer` = &quot;#FFC34D&quot;,
      `None` = &quot;#FFFFFF&quot;,
      `Transcription` = &quot;#008000&quot;,
      `Super-Enhancer` = &quot;#0000FF&quot; # This is blue, not used by ChromHMM!
    )
  
  # The bigwig tracks
  dtl &lt;- names(.bw) %&gt;%
    lapply(function(x){
      import(.bw[[x]], which = gr) %&gt;%
        DataTrack(
          name = x, type = &quot;l&quot;, 
          col = .bwCols[[x]],
          size = 1
        )
    })
  # Now form the plot
  list(
    ideo, .ax, hicTR, hmmTR
  ) %&gt;%
    c(h3K27acTR) %&gt;%
    c(
      subsetByOverlaps(gm, gr, ignore.strand = TRUE) %&gt;%
        GeneRegionTrack(
          name = &quot;Genes&quot;,
          transcriptAnnotation = &quot;symbol&quot;
          )
    ) %&gt;%
    c(dtl) %&gt;%
    plotTracks(
      from = start(gr),
      to = end(gr)
    )
}</code></pre>
<div id="the-most-highly-ranked-se" class="section level2">
<h2>The most highly ranked SE</h2>
<pre class="r"><code>gr &lt;- subset(seGR, rpmpbp_E2 == max(rpmpbp_E2))
gn &lt;- subsetByOverlaps(geneRanges, gr)</code></pre>
<p>Of all the putative super-enhancers, the most highly ranked in both conditions was the same. This was located on chr3 and had a width of 5584bp, being located within the overall range of FGF12. In addition, it appears this SE was only generated from one initial ChIP peak.</p>
<p>Whilst no differential expression was found for FGF12, it was expressed at low levels in T47D despite no discernible H3K27AC signal in it’s promoter region. The mid-gene super enhancer also appeared to connect with the genes own promoter.</p>
<pre class="r"><code>plotRegion(gr, zoom = 130)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/rose.Rmd/pltHighest-1.png" alt="*The most highly ranked super-enhancer in both treatments. HiC Interactions with the SE are shown using 10, 20, 40 &amp; 100kb bins combined, and are restricted to those within 1Mb. The ChromHMM model was generated using HMEC cells*" width="960" />
<p class="caption">
<em>The most highly ranked super-enhancer in both treatments. HiC Interactions with the SE are shown using 10, 20, 40 &amp; 100kb bins combined, and are restricted to those within 1Mb. The ChromHMM model was generated using HMEC cells</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pltHighest-1">
Past versions of pltHighest-1.png
</button>
</p>
<div id="fig-pltHighest-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/9ce4414c70630e544024bedfbcba90d73c704f92/docs/figure/rose.Rmd/pltHighest-1.png" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/0c4e66a99028ddc7edd459d2c36b16ddaecf05b9/docs/figure/rose.Rmd/pltHighest-1.png" target="_blank">0c4e66a</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/2491240e00ca86842fb5105c26622bbe5ba178ee/docs/figure/rose.Rmd/pltHighest-1.png" target="_blank">2491240</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="the-next-most-highly-ranked-se" class="section level2">
<h2>The next most highly ranked SE</h2>
<pre class="r"><code>gr &lt;- seGR %&gt;% 
  subset(rpmpbp_E2_DHT &lt; max(rpmpbp_E2_DHT)) %&gt;% 
  subset(rpmpbp_E2_DHT == max(rpmpbp_E2_DHT))
gn &lt;- subsetByOverlaps(geneRanges, gr)</code></pre>
<p>The next most highly ranked super-enhancer was located in an intergenic region. A HiC Interaction was detected between this SE and <em>DDX6</em>, although this was unlikely to be to the promoter of this gene. As this was a conserved super-enhancer, no logFC was observed at <em>DDX6</em> although the gene itself was highly expressed.</p>
<p>Despite the SE being located proximal to <em>CXCR5</em>, this gene was not detected as expressed in the RNA-Seq data.</p>
<pre class="r"><code>plotRegion(gr, zoom = 70, hic = gi)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/rose.Rmd/plotNext-1.png" alt="*The second most highly ranked super-enhancer. Connections were noted between this SE and DDX6.*" width="960" />
<p class="caption">
<em>The second most highly ranked super-enhancer. Connections were noted between this SE and DDX6.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotNext-1">
Past versions of plotNext-1.png
</button>
</p>
<div id="fig-plotNext-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/9ce4414c70630e544024bedfbcba90d73c704f92/docs/figure/rose.Rmd/plotNext-1.png" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/0c4e66a99028ddc7edd459d2c36b16ddaecf05b9/docs/figure/rose.Rmd/plotNext-1.png" target="_blank">0c4e66a</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/2491240e00ca86842fb5105c26622bbe5ba178ee/docs/figure/rose.Rmd/plotNext-1.png" target="_blank">2491240</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="a-super-enhancer-detected-in-e2-dht-only" class="section level2">
<h2>A Super Enhancer Detected in E2 + DHT Only</h2>
<pre class="r"><code>gr &lt;- seGR %&gt;% 
  subset(class == &quot;E2 + DHT Only&quot;) %&gt;%
  subset(rpmpbp_E2 == min(rpmpbp_E2))
zm &lt;- 1e3
gn &lt;- geneRanges %&gt;%
  subsetByOverlaps(
    resize(gr, width = zm*width(gr), fix = &quot;center&quot;), ignore.strand = TRUE
  ) %&gt;%
  subset(gene %in% rnaseq$gene_id)</code></pre>
<p>The most divergently ranked super enhancer was explored next, with a width of 4227bp. This was again located in an intergenic region, and was clearly lost in E2 samples, as expected. Whilst the signal in E2 + DHT was detectable, it is worth noting that this enhancer was still a relatively low-ranked SE in this treatment group.</p>
<p>No HiC connections were detected.</p>
<pre class="r"><code>plotRegion(gr, zoom = 100, hic = gi)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/rose.Rmd/plotDHT-1.png" alt="*Close view of the central super enhancer gained in E2 + DHT*" width="960" />
<p class="caption">
<em>Close view of the central super enhancer gained in E2 + DHT</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotDHT-1">
Past versions of plotDHT-1.png
</button>
</p>
<div id="fig-plotDHT-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/9ce4414c70630e544024bedfbcba90d73c704f92/docs/figure/rose.Rmd/plotDHT-1.png" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="the-largest-se" class="section level2">
<h2>The largest SE</h2>
<pre class="r"><code>gr &lt;- seGR %&gt;% subset(width == max(width))
zm &lt;- 8
gn &lt;- geneRanges %&gt;%
  subsetByOverlaps(resize(gr, width = zm*width(gr), fix = &quot;center&quot;), ignore.strand = TRUE) %&gt;%
  subset(gene %in% rnaseq$gene_id)</code></pre>
<p>The largest SE by size was found to be 131.297kb wide. Several long-range interactions were observed.</p>
<pre class="r"><code>plotRegion(
  gr, zoom = zm, 
  hic = gi, 
  # gm = subset(geneModels, gene %in% names(gn)), 
  maxIntDist = 1e7
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/rose.Rmd/plotLargest-1.png" alt="*View of the longest super enhancer detected in this dataset*" width="960" />
<p class="caption">
<em>View of the longest super enhancer detected in this dataset</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotLargest-1">
Past versions of plotLargest-1.png
</button>
</p>
<div id="fig-plotLargest-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/9ce4414c70630e544024bedfbcba90d73c704f92/docs/figure/rose.Rmd/plotLargest-1.png" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="the-longest-se-in-the-top-5-ranked-ses" class="section level2">
<h2>The longest SE in the top 5 ranked SEs</h2>
<pre class="r"><code>gr &lt;- subset(seGR, width == 19912)
gn &lt;- geneRanges %&gt;%
  subsetByOverlaps(resize(gr, width = zm*width(gr), fix = &quot;center&quot;), ignore.strand = TRUE) %&gt;%
  subset(gene %in% rnaseq$gene_id)</code></pre>
<p>This specific SE was 19.9kb long.</p>
<pre class="r"><code>plotRegion(gr, zoom = 100, hic = gi)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/rose.Rmd/plotLargestHigh-1.png" alt="*The longest SE in the top 5 ranked across both conditions*" width="960" />
<p class="caption">
<em>The longest SE in the top 5 ranked across both conditions</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotLargestHigh-1">
Past versions of plotLargestHigh-1.png
</button>
</p>
<div id="fig-plotLargestHigh-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/ENCODE_T47D_HiC_ENCSR549MGQ/blob/9ce4414c70630e544024bedfbcba90d73c704f92/docs/figure/rose.Rmd/plotLargestHigh-1.png" target="_blank">9ce4414</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<p>Whilst adhering to the ROSE approach, some of the identified super-enhancers appear to be somewhat less than compelling. Integration with other ChIP Seq data, or other data types may be prudent to confirm the status.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] magrittr_1.5                GenomicInteractions_1.24.0 
 [3] InteractionSet_1.18.0       SummarizedExperiment_1.20.0
 [5] Biobase_2.50.0              MatrixGenerics_1.2.0       
 [7] matrixStats_0.57.0          scales_1.1.1               
 [9] pander_0.6.3                Rsamtools_2.6.0            
[11] Biostrings_2.58.0           XVector_0.30.0             
[13] glue_1.4.2                  Gviz_1.34.0                
[15] forcats_0.5.0               stringr_1.4.0              
[17] dplyr_1.0.2                 purrr_0.3.4                
[19] readr_1.4.0                 tidyr_1.1.2                
[21] tibble_3.0.4                ggplot2_3.3.2              
[23] tidyverse_1.3.0             rtracklayer_1.50.0         
[25] GenomicRanges_1.42.0        GenomeInfoDb_1.26.0        
[27] IRanges_2.24.0              S4Vectors_0.28.0           
[29] BiocGenerics_0.36.0         workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1             backports_1.2.0          Hmisc_4.4-1             
  [4] BiocFileCache_1.14.0     igraph_1.2.6             lazyeval_0.2.2          
  [7] splines_4.0.3            BiocParallel_1.24.0      digest_0.6.27           
 [10] ensembldb_2.14.0         htmltools_0.5.0          fansi_0.4.1             
 [13] checkmate_2.0.0          memoise_1.1.0            BSgenome_1.58.0         
 [16] cluster_2.1.0            modelr_0.1.8             askpass_1.1             
 [19] prettyunits_1.1.1        jpeg_0.1-8.1             colorspace_1.4-1        
 [22] blob_1.2.1               rvest_0.3.6              rappdirs_0.3.1          
 [25] haven_2.3.1              xfun_0.19                crayon_1.3.4            
 [28] RCurl_1.98-1.2           jsonlite_1.7.1           survival_3.2-7          
 [31] VariantAnnotation_1.36.0 gtable_0.3.0             zlibbioc_1.36.0         
 [34] DelayedArray_0.16.0      DBI_1.1.0                Rcpp_1.0.5              
 [37] progress_1.2.2           htmlTable_2.1.0          foreign_0.8-79          
 [40] bit_4.0.4                Formula_1.2-4            htmlwidgets_1.5.2       
 [43] httr_1.4.2               RColorBrewer_1.1-2       ellipsis_0.3.1          
 [46] farver_2.0.3             pkgconfig_2.0.3          XML_3.99-0.5            
 [49] nnet_7.3-14              dbplyr_2.0.0             here_0.1                
 [52] labeling_0.4.2           tidyselect_1.1.0         rlang_0.4.8             
 [55] later_1.1.0.1            AnnotationDbi_1.52.0     munsell_0.5.0           
 [58] cellranger_1.1.0         tools_4.0.3              cli_2.1.0               
 [61] generics_0.1.0           RSQLite_2.2.1            broom_0.7.2             
 [64] evaluate_0.14            yaml_2.2.1               knitr_1.30              
 [67] bit64_4.0.5              fs_1.5.0                 AnnotationFilter_1.14.0 
 [70] nlme_3.1-149             whisker_0.4              xml2_1.3.2              
 [73] biomaRt_2.46.0           compiler_4.0.3           rstudioapi_0.11         
 [76] curl_4.3                 png_0.1-7                reprex_0.3.0            
 [79] stringi_1.5.3            highr_0.8                GenomicFeatures_1.42.0  
 [82] lattice_0.20-41          ProtGenerics_1.22.0      Matrix_1.2-18           
 [85] vctrs_0.3.4              pillar_1.4.6             lifecycle_0.2.0         
 [88] data.table_1.13.2        bitops_1.0-6             httpuv_1.5.4            
 [91] R6_2.5.0                 latticeExtra_0.6-29      promises_1.1.1          
 [94] gridExtra_2.3            dichromat_2.0-0          assertthat_0.2.1        
 [97] openssl_1.4.3            rprojroot_1.3-2          withr_2.3.0             
[100] GenomicAlignments_1.26.0 GenomeInfoDbData_1.2.4   mgcv_1.8-33             
[103] hms_0.5.3                rpart_4.1-15             rmarkdown_2.5           
[106] git2r_0.27.1             biovizBase_1.38.0        lubridate_1.7.9         
[109] base64enc_0.1-3         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
