---
title: "Check Super-Enhancers"
author: "Stephen Pederson<br>Dame Roma Mitchell Cancer Research Laboratories<br>Adelaide Medical School<br>University of Adelaide"
date: "`r format(Sys.Date(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

Given that Super-Enhancers have been called using the adaptation of the ROSE algorithm provided in the HOMER suite of tools, a robust check of these regions should be undertaken.
Enhancers and super-enhancers were called from the raw alignments provided in the H3K27AC ChIP-Seq dataset, available in the primary repository at `data/external/H3K27AC/bam`.
Notably these files are not in the smaller working repositories.

Traces of these alignments are provided in `data/external/H3K27AC/bigwig` and these will be used to inspect the super-enhancer calls

```{r}
library(rtracklayer)
library(Gviz)
library(tidyverse)
library(magrittr)
library(yaml)
```


```{r config}
config <- here::here("config/config.yml") %>%
  read_yaml()
```



## Import GeneModels

```{r gm}
geneModels <- read_rds(
  here::here("output/geneModels.rds")
)
```



## Define the Common Seqinfo Object

```{r sq}
sq <- seqinfo(geneModels)
```


## Load BigWig Files

```{r}
bw <- here::here("data/external/H3K27AC/bigwig") %>%
  list.files(pattern = "(_[1-3]|input).bigwig", full.names = TRUE) %>%
  BigWigFileList() %>%
  setNames(
    str_replace_all(path(.), ".+T47D_(.+).bigwig", "\\1")
  )
bwCols <- case_when(
  grepl("DHT", names(bw)) ~"red", 
  grepl("Pooled", names(bw)) ~ "grey",
  grepl("E2_[1-3]", names(bw)) ~ "blue"
) %>%
  setNames(names(bw))
```

## Define Enhancers

```{r}
enh_cols <- c("PeakID", "chr", "start", "end", "strand")
enhGR <- here::here("output/HOMER") %>%
  list.files(pattern = "^enhanc", full.names = TRUE, recursive = TRUE) %>%
  sapply(read_tsv, comment = '#', simplify = FALSE, col_names = enh_cols, col_types = c("ccnnc------")) %>%
  setNames(dirname(names(.))) %>%
  setNames(str_extract(names(.), "E2.*")) %>%
  lapply(function(x){
    makeGRangesFromDataFrame(x, keep.extra.columns = TRUE, seqinfo = sq)
  }) %>%
  lapply(sort)
enhGR <- names(enhGR) %>%
  lapply(function(x, gr = enhGR){
    
    df <- with(
      mcols(gr[[x]]),
      DataFrame(
        type = "exon",
        gene = PeakID,
        exon = PeakID,
        transcript = PeakID,
        symbol = x
      )
    )
    mcols(gr[[x]]) <- df
    gr[[x]]
  }) %>%
  as("GRangesList") %>%
  unlist() %>%
  sort()
# supGR <- here::here("output/HOMER") %>%
#   list.files(pattern = "^superEnhanc", full.names = TRUE, recursive = TRUE) %>%
#   sapply(read_tsv, comment = '#', simplify = FALSE, col_names = enh_cols, col_types = c("ccnnc------")) %>%
#   setNames(basename(dirname(names(.)))) %>%
#   lapply(function(x){
#     makeGRangesFromDataFrame(x, keep.extra.columns = TRUE, seqinfo = sq)
#   }) %>%
#   lapply(sort)
# supGR <- 
# http://www.licpathway.net/sedb/download/TE_bed/Sample_02_401_TE.bed
here::here("data/external/Sample_02_401_TE.bed") %>%
  import.bed15()
supGR <- names(supGR) %>%
  lapply(function(x, gr = supGR){
    
    df <- with(
      mcols(gr[[x]]),
      DataFrame(
        type = "exon",
        gene = PeakID,
        exon = PeakID,
        transcript = PeakID,
        symbol = x
      )
    )
    mcols(gr[[x]]) <- df
    gr[[x]]
  }) %>%
  as("GRangesList") %>%
  unlist() %>%
  sort()
```

## Swap Enhancers to BED Files

Just to a temp switch out of the HOMER SE Results for the H3K27ac ChIP Peaks

```{r}
chipGR <- here::here("data/external/BED") %>%
  list.files(pattern = "H3K27", full.names = TRUE) %>%
  sapply(import.bed, seqinfo = sq) %>%
  lapply(sort) %>%
  setNames(
    str_replace(names(.), ".+(E2.*)_peaks.bed", "\\1")
  )
chipGR <- names(chipGR) %>%
  lapply(function(x, gr = chipGR){
    
    df <- DataFrame(
        type = "exon",
        gene = paste0(seqnames(gr[[x]]), "-", start(gr[[x]])),
        exon = paste0(seqnames(gr[[x]]), "-", start(gr[[x]])),
        transcript = paste0(seqnames(gr[[x]]), "-", start(gr[[x]])),
        symbol = x
    )
    mcols(gr[[x]]) <- df
    gr[[x]]
  }) %>%
  as("GRangesList") %>%
  unlist() %>%
  sort()
```


## Track Setup

Any tracks which remained independent of peaks and genes were also setup

```{r}
gen <- "hg19"
ax <- GenomeAxisTrack()
```


## Define the Range to Plot


## Plot Range

```{r}
plotGeneRange <- function(g, zoom = 3, gm = geneModels, enh1 = enhGR, enh2 = chipGR, .bw = bw, .bwCols = bwCols, .ax = ax, gen = "hg19"){
  
  # Set the range for the gene
  gr <- subset(gm, symbol == g)
  gr <- range(gr)
  stopifnot(length(gr) > 0)
  gr <- resize(range(gr), width = zoom*width(gr), fix = "center")
  # Form the Ideogram track
  chr <- unique(as.character(seqnames(gr)))
  ideo <- IdeogramTrack(genome = gen, chromosome = chr)
  # The enhancer track (may be something else really)
  enh1TR <- subsetByOverlaps(enh1, gr, ignore.strand = TRUE) %>%
    GeneRegionTrack(name = "HOMER", transcriptAnnotation = "symbol", size = 1.5, fill = "red")
  # The super enhancer track
  enh2TR <- subsetByOverlaps(enh2, gr, ignore.strand = TRUE) %>%
    GeneRegionTrack(name = "ChIP", transcriptAnnotation = "symbol", size = 1.5, fill = "green")
  # The bigwig tracks
  dtl <- names(.bw) %>%
    lapply(function(x){
      import(.bw[[x]], which = gr) %>%
        DataTrack(
          name = x, type = "l", 
          col = .bwCols[[x]]
        )
    })
  # Now form the plot
  list(
    ideo, .ax
  ) %>%
    c(enh1TR, enh2TR) %>%
    c(
      subsetByOverlaps(gm, gr, ignore.strand = TRUE) %>%
        GeneRegionTrack(name = "Genes",transcriptAnnotation = "symbol")
    ) %>%
    c(dtl) %>%
    plotTracks(
      from = start(gr),
      to = end(gr)
    )
  
}
```


```{r}
plotGeneRange("FOXP3", 3)
```



In conclusion:

- Too many of these overlap promoters and should be excluded
- The Enhancers file exclude anything considered as a SuperEnhancer

The solution may be to:

- Find all Enhancers then exclude those that overlap a TSS. This can be done using `findPeaks --style histone` and excluding promoters in R. The remaining regions detected should be the enhancers which are candidates for SE status.
- Use this list of Enhancers (minus promoters) to filter the BAM files
- Submit these for SuperEnhancer Identification
