---
title: "Integration of HiC and Super Enhancer Data"
author: "Stephen Pederson<br>Dame Roma Mitchell Cancer Research Laboratories<br>Adelaide Medical School<br>University of Adelaide"
date: "`r format(Sys.Date(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center",
  fig.width = 10,
  fig.height = 10
)
```


```{r packages}
library(rtracklayer)
library(tidyverse)
library(Gviz)
library(glue)
library(Rsamtools)
library(pander)
library(scales)
library(GenomicInteractions)
library(InteractionSet)
library(magrittr)
library(plyranges)
library(goseq)
```

```{r setOpts}
theme_set(theme_bw())
panderOptions("big.mark", ",")
panderOptions("table.split.table", Inf)
nc <- ceiling(detectCores()/2)
```


# Introduction

## Data Description

Given the groups of peaks identified by Leila Hosseinzadeh, using ChIP-Seq from both GATA3 and the AR, the connection of these peaks to enhancers, promoters and DE genes was explored.
The groupings were provided as:

1. GATA3 sites detected under DHT treatment only
2. Shared sites between GATA3 and AR which are gained under DHT treatment
3. AR sites which were independent of DHT treatment
4. AR sites detected under DHT treatment only
5. ... (I'll finish later)

Groups defining the relationship between AR and GATA3 at each peak were specified in the following files.

```{r groupBed}
groupBed <- here::here("data/external/BED/AR_GATA3_Groups") %>%
  list.files(full.names = TRUE, pattern = "bed")
groupBed %>%
  basename() %>%
  enframe(name = c(), value = "filename") %>%
  rename_all(str_to_title) %>%
  pander(
    justify = "l",
    caption = "List of `.bed` files containing peaks for each of the defined groups"
  )
```

These groups were defined from consensus peaks within each treatment across replicates.
The bed files associated with these treatment-specific consensus sites were all obtained using Galaxy as part of her PhD candidature.
Files used for generation of consensus peaks are given in the table below.

```{r chipBed}
chipBed <- here::here("data/external") %>%
  list.files(full.names = TRUE, pattern = "bed", recursive = TRUE) %>%
  str_subset("ChromHMM|Groups", TRUE)
```


```{r bedTable}
chipBed %>%
  basename() %>%
  enframe(name = c(), value = "filename") %>%
  mutate(
    `Cell Type` = str_extract(filename, "T47D"),
    target = str_extract(filename, "AR|GATA3|H3K27Ac"),
    treatment = case_when(
      str_detect(filename, "E2_DHT") ~ "E2+DHT",
      str_detect(filename, "E2") ~ "E2"
    )
  ) %>%
  arrange(target, treatment) %>%
  rename_all(str_to_title) %>%
  pander(
    justify = "llll",
    caption = "Description of `.bed` files used for defining consensus peaks."
  )
```


## External data

**ENCODE HiC data** for T47D cells was analysed as part of the workflow.
Significant interactions were detected using bins of 10, 20, 40 and 100kb.

**RNASeq data** was also incorporated for T47D cells using Stripped media.
A comparison between DHT treatment and Vehicle was performed.
It should be noted that this dataset was generated using totalRNA and some technical issues appear to have reduced the statistical power of this dataset.

**H3K27Ac data** from both E2 treatment and E2 + DHT was incorporated.
Peaks were divided into promoters, based on overlap with any TSS from an expressed gene in the RNA-Seq data, and enhancers.
Enhancers were considered to be those peaks which didn't overlap any TSS, and super-enhancers were detected using the ROSE algorithm without modification.


## Analytic Approaches

```{r minWidth}
minWidth <- 150
```


All analyses of ChIP-Seq data was performed using consensus peaks for each transcription factor or histone mark.
To form consensus peaks, all peaks were combined across treatments and overlapping peaks were merged into a single peak, encompassing the combined range.
In addition, any putative peaks $\leq$`r minWidth`bp were excluded as low quality peaks.


# Data Import

## Gene-Level Data

```{r rnaseq}
rnaseq <- here::here("data/external/RNASeq/t47d_DHT_StrippedSerum_RNASeq_topTable.tsv") %>%
  read_tsv()
```

```{r geneModels}
geneModels <- here::here("output/geneModels.rds") %>%
  read_rds()
geneRanges <- geneModels %>% 
  split(f = .$gene) %>% 
  range() %>% 
  unlist() %>%
  sort()
geneRanges$gene <- names(geneRanges)
geneRanges$symbol <- geneModels$symbol[match(geneRanges$gene, geneModels$gene)]
sq <- seqinfo(geneModels)
transModels <- here::here("output/transModels.rds") %>%
  read_rds()
```

Genes were parsed as condensed models just containing all relevant exons, and the ranges defined by the start/end points of each gene were also defined for easy comparison.

The RNA-Seq results from Stripped Serum were also loaded.
This dataset comes with some caveats, in that very few genes were detected as DE in response to DHT treatment.
This was likely to be an artefact of poor rRNA removal and the biases this leaves within a dataset.
In total this dataset contained `r comma(nrow(rnaseq))` detectable genes, broken into `r nrow(dplyr::filter(rnaseq, DE & logFC > 0))` up-regulated genes, `r nrow(dplyr::filter(rnaseq, DE & logFC < 0))` down-regulated genes, with the remaining `r comma(nrow(dplyr::filter(rnaseq, !DE)))` genes considered detectable, but not differentially expressed.

Whilst the comparison of AR and GATA3 ChIP peaks with differential expression may yield some insights, the clarification of the potential consequences for each peak should still be able to be extrapolated from the HiC and H3K27Ac data.

## ChIP-Seq Data

```{r allPeaks}
allPeaks <- chipBed %>%
    sapply(function(x){
    import.bed(x) %>%
      subset(seqnames %in% seqlevels(sq)) %>%
      sortSeqlevels() %>%
      keepSeqlevels(value = seqlevels(sq)) %>%
      `seqinfo<-`(value = sq) %>%
      sort
  }, simplify = FALSE) %>%
  setNames(basename(names(.))) %>%
  setNames(str_remove_all(names(.), "^.+T47D.")) %>%
  setNames(str_extract_all(names(.), "^[A-Za-z0-9_\\-]+")) %>%
  setNames(str_remove_all(names(.), "_peaks.*"))
```

```{r allWidths}
allWidths <- names(allPeaks) %>%
  lapply(
    function(x){
      tibble(width = width(allPeaks[[x]]), sample = x)
    }
  ) %>%
  bind_rows() %>%
  mutate(
    target = str_extract(sample, "AR|GATA3|H3K27Ac"),
    treatment = case_when(
      str_detect(sample, "E2_DHT") ~ "E2+DHT",
      str_detect(sample, "E2") ~ "E2",
    )
  ) %>%
  arrange(sample, width) %>%
  group_by(sample, target, treatment) %>%
  mutate(n = seq_along(width) / length(width)) 
```

All previously identified ChIP-Seq peaks were assessed for width, given that peaks of 1bp were noticed in all samples.
Peaks which are too short are unlikely to be true peaks and were removed,
In addition, it was noticed that:

- H3K27Ac peaks were much broader than either GATA3 or AR peaks
- AR peaks in the presence of DHT appeared to be broader. This may be a technical artefact. In general peak widths are dependent on fragment size defined prior to sequencing, and it is likely that the fragments in these samples were slightly longer than in the E2 alone samples.
- GATA3 peak widths bore a close resemblance to each other and to Ar peak widths in E2 alone.


```{r plotWidths, fig.width = 7, fig.height=7, fig.cap=glue("*Assessment of peak width for each set of consensus peaks. H3K27Ac peaks were noticeably wider than GATA3 or AR peaks, with AR peaks also appearing wider in the presence of DHT. The suggested peak size of {minWidth}bp is shown as a vertical line.*")}
allWidths %>%
  ggplot(aes(width, n, colour = sample)) +
  geom_line() +
  geom_vline(xintercept = minWidth, linetype = 2) +
  scale_x_log10(breaks = 10^seq(1, 5), labels = comma, expand = expansion(0, 0)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = percent, expand = expansion(c(0.02, 0.02))) +
  scale_colour_brewer(palette = "Paired") +
  labs(
    x = "Peak Width",
    y = "Cumulative Fraction",
    colour = "Sample"
  ) +
  theme(
    legend.position = c(0.01, 0.99),
    legend.justification = c(0, 1)
  )
```

```{r tabAllWidths}
allWidths %>% 
  summarise(
    Retained = sum(width > minWidth),
    Lost = sum(width <= minWidth),
    `% Lost` = mean(width <= minWidth)
    ) %>%
  mutate(`% Lost` = percent(`% Lost`, accuracy = 0.1)) %>%
  rename_all(str_to_title) %>%
  pander(
    justify = "lllrrr",
    caption = glue(
      "Summary of ChIP-Seq peaks for each target after restricting to a minimum peak size >{minWidth}bp."
    )
  )
```

```{r removeSmallPeaks}
allPeaks <- lapply(allPeaks, subset, width > minWidth)
```

In addition to the peaks, connections to the bigwig files were also setup for the purposes of visualisation.

```{r bw}
bw <- here::here("data/external") %>%
  list.files(pattern = "bigwig", full.names = TRUE, recursive = TRUE) %>%
  str_subset("_[1-3].bigwig", TRUE) %>%
  str_subset("Pooled", TRUE) %>%
  str_subset("H3K.+merged", TRUE) %>%
  BigWigFileList() 
names(bw) <- path(bw) %>%
  basename() %>%
  str_remove_all("[\\[\\]]") %>%
  str_replace_all("[-\\+]", "_") %>%
  str_replace_all(".*T47D_(GATA3|AR|H3K27Ac)_(.+)(_merged.bigwig|.bigwig)", "\\1_\\2") %>%
  str_remove_all("_merged")
```



## HiC Interactions

Given that the different resolutions enabled by the different bin sizes will yield differing powers to detect interactions, significant (i.e. above background) interactions were loaded at 3 different resolutions: 10kb, 20kb and 40kb.

```{r gi}
gi40 <- here::here("output/gi_40000.rds") %>% read_rds()
gi20 <- here::here("output/gi_20000.rds") %>% read_rds()
gi10 <- here::here("output/gi_10000.rds") %>% read_rds()
gi <- c(gi10, gi20, gi40) %>%
  sort()
gi$binSize <- width(gi)$first
```

```{r tabInteractions}
gi %>%
  anchorOne() %>%
  width() %>% 
  table() %>% 
  enframe(name = "Bin Size", value = "Number of Interactions") %>% 
  mutate_all(as.numeric) %>%
  pander(
    caption = "Breakdown of cis interactions detected for each bin size in the Encode T47D HiC data."
  )
```

## H3K27Ac Data

```{r parseH3K27Ac}
promGR <- here::here("output/promoters.bed") %>% 
  import.bed() %>%
  `seqinfo<-`(value = sq) %>%
  granges()
promGR$class <- "Promoter"
enhGR <- here::here("output/enhancers.bed") %>% 
  import.bed() %>%
  `seqinfo<-`(value = sq) %>%
  granges()
enhGR$class <- "Enhancer"
seGR <- here::here("output/SE.gtf") %>%
  import.gff() %>%
  `seqinfo<-`(value = sq) %>%
  granges()
seGR$class <- "Super Enhancer"
h3k27ac <- GRangesList(
  promGR,
  enhGR,
  seGR
) %>%
  unlist() %>%
  sort() %>% 
  setNames(paste(seqnames(.), start(.), sep = "_")) %>% 
  names_to_column("peak_id") %>%
  setNames(paste(.$class, .$peak_id, sep = "_")) %>% 
  names_to_column("feature_id")
```

```{r updateH3K27Ac}
h3k27ac$E2 <- rep(FALSE, length(h3k27ac))
h3k27ac$E2[queryHits(findOverlaps(h3k27ac, allPeaks$H3K27Ac_E2))] <- TRUE
h3k27ac$E2_DHT <- rep(FALSE, length(h3k27ac))
h3k27ac$E2_DHT[queryHits(findOverlaps(h3k27ac, allPeaks$H3K27Ac_E2_DHT))] <- TRUE
h3k27ac$group <- mcols(h3k27ac) %>%
  with(
    case_when(
      E2 & E2_DHT ~ "Common",
      !E2_DHT & E2 ~ "E2 Only",
      E2_DHT & !E2 ~ "E2+DHT Only",
      !E2_DHT & !E2 ~ "None"
    )
  )
h3k27ac <- subset(h3k27ac, group != "None")
```


It should also be noted that in the previous steps, promoters were restricted to H3K27Ac peaks which overlapped the TSS of genes detected as expressed in the RNA-Seq dataset.
Any peaks which overlapped the TSS/Promoter of an undetectable gene were classed as Enhancers.

```{r tabH3K27Ac}
h3k27ac %>%
  with(
    tibble(
      width = width,
      class = class
    )
  ) %>%
  mutate(
    class = factor(class, levels = c("Promoter", "Enhancer", "Super Enhancer"))
  ) %>%
  group_by(class) %>%
  summarise(
    Total = dplyr::n(),
    `Total (bp)` = sum(width),
    `Median Width` = median(width),
    Shortest = min(width),
    Longest = max(width)
  ) %>%
  rename_all(str_to_title) %>%
  pander(
    caption = "Description of all features defined by the H3K27Ac dataset. Any Enhancers contained within Super Ehancers were retained in both features"
  )
```

Information regarding which dataset contained an overlap with each consensus peak was then added to each H3K27Ac site.

```{r tabH3K27AcSummary}
h3k27ac %>%
  subset(E2_DHT | E2) %>% # Some peaks may have been removed as < minWidth, but pass after merging
  mcols() %>%
  as.data.frame() %>%
  group_by(class, E2_DHT, E2) %>%
  tally() %>%
  ungroup() %>%
  pivot_wider(names_from = class, values_from = n, values_fill = 0) %>%
  mutate(
    `Treatment Group` = case_when(
      E2 & E2_DHT ~ "Both",
      !E2 & E2_DHT ~ "DHT Only",
      E2 & !E2_DHT ~ "E2 Only"
    )
    ) %>%
  arrange(`Treatment Group`) %>%
  dplyr::select(`Treatment Group`, Promoter, contains("Enhancer")) %>%
  pander(
    caption = "
    *Summary of consensus H3K27Ac features by treatment group, simply showing presence/absence, not accounting for changed binding intensity.
    Notably, all Super Enhancers contained peaks which overlapped them in each group, however this does __not__ ensure that all Super Enhancers were defined as Super Ehancers in both treatment groups.
    The ROSE workflow did detect a small number of SEs which were unique to DHT treatment.*
    "
  )
```

# All ChIP Data

Both GATA3 and AR ChIP datasets were loaded and consensus peaks identified based on the total size of any overlapping regions.
These were then annotated simply based on gain or loss in DHT treatment, noting that GATA3 data contains two conditions (*Vehicle* and *DHT*) whilst AR data contained the conditions *E2* and *E2+DHT*.
BigWig files contain the more subtle information regarding coverage, whilst this classification can indicate complete loss or gain of binding.

## All GATA3 Sites

```{r gata3ConsGR}
gata3ConsGR <- allPeaks %>% 
  .[str_detect(names(.), "GATA3")] %>%
  as("GRangesList") %>%
  unlist() %>%
  GenomicRanges::reduce() %>%
  setNames(paste0(seqnames(.), "_", start(.))) %>%
  names_to_column("peak_id")
gata3ConsGR$E2 <- seq_along(gata3ConsGR) %in% queryHits(
  findOverlaps(gata3ConsGR, allPeaks$`GATA3-E2`)
)
gata3ConsGR$E2_DHT <- seq_along(gata3ConsGR) %in% queryHits(
  findOverlaps(gata3ConsGR, allPeaks$`GATA3-E2_DHT`)
)
gata3ConsGR$group <- mcols(gata3ConsGR) %>%
  with(
    case_when(
      E2 & E2_DHT ~ "Common",
      !E2 & E2_DHT ~ "E2+DHT Only",
      E2 & !E2_DHT ~ "E2 Only"
    )
  )
```

In order to characterise the behaviours of the defined groups, the consensus set of `r comma(length(gata3ConsGR))` GATA3 peaks was defined as those detected in either Vehicle or DHT.
In keeping with all steps of this analysis, any peaks $\leq `r minWidth`$bp were excluded.

```{r tabGata3ConsSummary}
gata3ConsGR$group %>%
  table() %>%
  enframe(name = "Group", value = "Peaks") %>%
  mutate(Peaks = as.numeric(Peaks)) %>%
  bind_rows(
    tibble(Group = "Total", Peaks = sum(.$Peaks))
  ) %>%
  pander(
    caption = glue(
      "Summary of consensus GATA3 peaks, after exclusion of any peaks $\\leq${minWidth}bp."
    ),
    emphasize.strong.rows = which(str_detect(.$Group, "Total"))
  )
```


```{r gata3Cons2Feature}
gata3Cons2Feature <- gata3ConsGR %>% 
  join_overlap_inner(h3k27ac) %>% 
  mcols() %>% 
  split(f = .$peak_id.x) %>% 
  mclapply(function(x){unique(x$class)}, mc.cores = nc)
```

Each consensus peak was then mapped to the features which overlapped the peak for downstream enrichment testing.
A total of `r comma(length(gata3Cons2Feature))` peaks were mapped to `r comma(sum(vapply(gata3Cons2Feature, length, integer(1))))` features, noting that many enhancers will overlap super enhancers.

## All AR Sites

```{r arConsGR}
arConsGR <- allPeaks %>%
  .[str_detect(names(.), "AR")] %>%
  as("GRangesList") %>%
  unlist() %>%
  GenomicRanges::reduce() %>%
  setNames(paste0(seqnames(.), "_", start(.))) %>%
  names_to_column("peak_id") 
arConsGR$E2 <- seq_along(arConsGR) %in% queryHits(findOverlaps(arConsGR, allPeaks$AR_E2))
arConsGR$E2_DHT <- seq_along(arConsGR) %in% queryHits(findOverlaps(arConsGR, allPeaks$AR_E2_DHT))
arConsGR$group <- mcols(arConsGR) %>%
  with(
    case_when(
      E2 & E2_DHT ~ "Common",
      !E2 & E2_DHT ~ "E2+DHT Only",
      E2 & !E2_DHT ~ "E2 Only",
      !E2 & !E2_DHT ~ "None"
    )
  )
```

In order to characterise the behaviours of the defined groups, the consensus set of `r comma(length(arConsGR))` AR peaks was defined as those detected in either Vehicle or DHT, again relying on bigwig files for more subtle information regarding changes in coverage.

```{r tabArConsSummary}
arConsGR$group %>%
  table() %>%
  enframe(name = "Group", value = "Peaks") %>%
  mutate(Peaks = as.numeric(Peaks)) %>%
  bind_rows(
    tibble(Group = "Total", Peaks = sum(.$Peaks))
  ) %>%
  pander(
    caption = glue(
      "Summary of consensus AR peaks, after exclusion of any peaks $\\leq${minWidth}bp."
    ),
    emphasize.strong.rows = which(str_detect(.$Group, "Total"))
  )
```



```{r arCons2Feature}
arCons2Feature <- arConsGR %>% 
  join_overlap_inner(h3k27ac) %>% 
  mcols() %>% 
  split(f = .$peak_id.x) %>% 
  mclapply(function(x){unique(x$class)}, mc.cores = nc)
```

Each consensus peak was then mapped to the features which overlapped the peak.
A total of `r comma(length(arCons2Feature))` peaks were mapped to `r comma(sum(vapply(arCons2Feature, length, integer(1))))` features, noting that many enhancers will overlap super enhancers.


# ChIP Peaks By AR & GATA3 Behavioural Groups


## Plotting Setup

```{r plotRegion}
gen <- "hg19"
ax <- GenomeAxisTrack()
plotRegion <- function(gr, zoom = 1, hic = gi, h3k27 = h3k27ac, gata3 = gata3ConsGR, ar = arConsGR, bigwig = bw, gm = geneModels, .ax = ax, rna = rnaseq, shift = 0, fs = 10, highlight = TRUE){
  
  stopifnot(is(gr, "GRanges") | length(gr) > 0)
  
  ## Form the Ideogram track
  chr <- unique(as.character(seqnames(gr)))
  ideo <- IdeogramTrack(genome = gen, chromosome = chr)
  
  ## The HiC Track
  goiInteractions <- hic %>%
    subsetByOverlaps(gr, ignore.strand = TRUE) %>%
    as("GenomicInteractions")
  hicTR <- InteractionTrack(
    goiInteractions,
    chromosome = chr,
    name = "HiC"
  )
  displayPars(hicTR) <- list(fontsize = fs, size = 1.2)
  
  ## Expand the range after looking for interactions within the initial range
  ## This keeps the range of interest in the centre
  origGR <- gr
  gr <- resize(gr, width = zoom*width(gr), fix = "center")
  gr <- shift(gr, shift)

  ## Setup the promoters, enhancers and super-enhancers as separate tracks
  promTR <- h3k27 %>%
    subsetByOverlaps(gr, ignore.strand = TRUE) %>%
    subset(class %in% c("Promoter", "Enhancer")) %>%
    GeneRegionTrack(
      name = "H3K27Ac",
      stacking = "dense",
      col = "transparent",
      feature = .$group,
      fontsize = fs,
      `Common` = "grey",
      `E2+DHT Only` = "red",
      `E2 Only` = "blue",
      size = 0.8
    )
  if (length(promTR) == 0) promTR <- c()
  h3CovGR <- names(bigwig) %>%
    str_subset("H3K27Ac") %>%
    lapply(function(x){
      gr <- import.bw(bigwig[[x]], which = gr)
      names(mcols(gr)) <- str_remove(x, "H3K27Ac_") 
      gr
    }) %>%
    as("GRangesList") %>%
    unlist()
  h3CovTR <- h3CovGR %>%
    DataTrack(
      groups = names(bigwig) %>%
        str_subset("H3K27Ac") %>%
        str_remove( "H3K27Ac_"),
      name = "H3K27Ac", type = "l", size = 1,
      legend = FALSE,
      col = c("blue", "red"),
      size = 2,
      fontsize = fs
  )
  seTR <- h3k27 %>%
    subsetByOverlaps(gr, ignore.strand = TRUE) %>%
    subset(class == "Super Enhancer") %>%
    GeneRegionTrack(
      name = "SE",
      stacking = "dense",
      col = "transparent",
      fill = "#FFC34D",
      fontsize = fs,      
      size = 0.7
    )
  if (length(seTR) == 0) seTR <- c()
  
  ## Setup the coverage track for GATA3
  gata3CovGR <- names(bigwig) %>%
    str_subset("GATA3") %>%
    lapply(function(x){
      gr <- import.bw(bigwig[[x]], which = gr)
      names(mcols(gr)) <- str_remove(x, "GATA3_") 
      gr
    }) %>%
    as("GRangesList") %>%
    unlist()
  gata3CovTR <- gata3CovGR %>%
    DataTrack(
      groups = names(bigwig) %>%
        str_subset("GATA3") %>%
        str_remove( "GATA3_"),
      name = "GATA3", type = "l", size = 1,
      legend = FALSE,
      col = c("blue", "red"),
      size = 2,
      fontsize = fs,
  )
  gata3TR <-  gata3 %>%
    subsetByOverlaps(gr, ignore.strand = TRUE) %>%
      GeneRegionTrack(
      name = "GATA3",
      stacking = "dense",
      feature = .$group,
      size = 0.8,
      col = "transparent",
      fontsize = fs,
      `Common` = "grey",
      `E2+DHT Only` = "red",
      `E2 Only` = "blue"
    )
  
  arTR <- ar %>%
    subsetByOverlaps(gr, ignore.strand = TRUE) %>%
    GeneRegionTrack(
      name = "AR",
      stacking = "dense",
      feature = .$group,
      size = 0.8,
      col = "transparent",
      fontsize = fs,      
      `Common` = "grey",
      `E2+DHT Only` = "red",
      `E2 Only` = "blue"
    )
  
  arCovGR <- names(bigwig) %>%
    str_subset("AR") %>%
    lapply(function(x){
      gr <- import.bw(bigwig[[x]], which = gr)
      names(mcols(gr)) <- str_remove(x, "AR") 
      gr
    }) %>%
    as("GRangesList") %>%
    unlist()
  arCovTR <- arCovGR %>%
    DataTrack(
      groups = names(bigwig) %>%
        str_subset("AR") %>%
        str_remove( "AR"),
      name = "AR", type = "l", size = 1,
      legend = FALSE,
      col = c("blue", "red"),
      size = 2,
      fontsize = fs
  )
  
  olGenes <- unique(subsetByOverlaps(gm, gr, ignore.strand = TRUE)$gene)
  expTR <- gm %>%
    subset(gene %in% olGenes) %>%
    subset(gene %in% rna$gene_id) %>%
    GeneRegionTrack(
      name = "Expressed",
      transcriptAnnotation = "symbol",
      fill = "#008000",
      col = "transparent",
      fontsize = fs
    )
  notExpGR <- gm %>%
    subset(gene %in% olGenes) %>%
    subset(!gene %in% rna$gene_id)
  notExpTR <- c()
  if (length(notExpGR) > 0){
    notExpTR <- GeneRegionTrack(
      notExpGR,
      name = "Not Expressed",
      transcriptAnnotation = "symbol",
      fill = rgb(0.5, 0.5, 0.5, 0.5),
      size = 0.8,
      fontsize = fs,  
      collapseTranscripts = "meta"
    )
  }
  
  ## A rectangle highlight around the original range
  highTR <- c(gata3TR, gata3CovTR, arTR, arCovTR, promTR, h3CovTR, expTR)
  if (highlight) {
    highTR <- HighlightTrack(
      trackList = highTR,
      range = resize(origGR, width = max(1e-2*zoom*width(origGR), width(origGR)), fix = "center"),
      # col = rgb(0.9, 0.7, 0),
      col = "black",
      fill = "#FFFFFF00",
      inBackground = FALSE
    )  
  }
  
  # Now form the plot
  c(
    ideo, .ax, hicTR, seTR
  ) %>%
    c(highTR) %>%
    c(notExpTR) %>%
    plotTracks(
      from = start(gr),
      to = end(gr)
    )
}
```

As much of the group-level analysis will involve plotting, a simple wrapper function was written to enable simple plotting of all tracks.


## GATA3 Sites Gained With DHT

```{r chipGR}
chipGR <- here::here("data/external/BED/AR_GATA3_Groups/Galaxy280-[T47D-GATA3-DHT-ONLY].bed") %>%
  import.bed() %>%
  subset(seqnames %in% seqlevels(sq)) %>%
  sortSeqlevels() %>%
  keepSeqlevels(value = seqlevels(sq)) %>%
  sort() %>%
  subset(width > minWidth) %>%
  c(
    subsetByOverlaps(gata3ConsGR, .)
  ) %>%
  GenomicRanges::reduce() %>%
  setNames(paste0(seqnames(.), "_", start(.))) %>%
  names_to_column("peak_id") %>%
  `seqinfo<-`(value = sq)
```

The first group of sites to integrate were the GATA3 ChIP peaks gained under DHT treatment, but where no corresponding increase was seen in the AR ChIP peaks.
Data was imported for a total of `r comma(length(chipGR))` peaks, excluding peaks $\leq$ `r minWidth`bp.
This peaks were merged with the consensus GATA3 peaks defined previously to ensure direct comparability between groups.
This set of peaks captured `r percent(length(chipGR) / length(gata3ConsGR))` of consensus GATA3 peaks.

```{r GATA3_ol_H3K27}
ol <- findOverlaps(chipGR, h3k27ac)
```

The first step is to subset and merge these sites with those which overlap a defined promoter or enhancer.
This step reduced the set of peaks to `r comma(length(GenomicRanges::reduce(chipGR)))` unique peaks,
however, given that some peaks may partially overlap multiple H3K27Ac-defined regions, this represented a total of `r comma(length(chipGR))` mappings between the two datasets.

```{r tabChipRMapping}
chipGR %>%
  join_overlap_inner(h3k27ac) %>%
  with(
    tibble(
      width = width,
      class = class,
      feature_id = feature_id
    )
  ) %>%
  group_by(class) %>%
  summarise(
    `Total Mapped Peaks` = dplyr::n(),
    `Total (bp)` = sum(width),
    `Median Width` = median(width),
    Shortest = min(width),
    Longest = max(width)
  ) %>%
  rename_all(str_to_title) %>%
  pander(
    caption = glue(
      "Summary of mappings from ChIP peaks (DHT-gained GATA3 peaks) to regions defined by the H3K27Ac dataset.",
      "Of the original {comma(length(chipGR))} peaks in this group, only {comma(sum(.$`Total Mapped Peaks`))} were mapped to a regulatory feature."
    )
  )
```


### Enrichment Testing

The first step is to ascertain whether the peaks where GATA3 binding is increased under DHT treatment, are associated with any specific feature from the H3K27Ac dataset, such as Promoters, Enhancers of Super Enhancers.
The size of the peak may bias any mapping and this was used as a sampling offset using the Wallenius approximation for biased sampling in the hypergeometric distribution.

For enrichment testing, each feature was considered as analogous to a gene, with a peak being mapped to it being considered as 'success'.
All 3 feature types were assessed by considering that feature type, and grouping the other feature types.
In this manner, only the peaks which overlapped a defined feature were considered for enrichment testing.

```{r goseqGata3}
mapped <- seq_along(gata3ConsGR) %in% queryHits(findOverlaps(gata3ConsGR, chipGR)) %>%
  setNames(gata3ConsGR$peak_id)
pwf <- nullp(mapped, bias.data = width(gata3ConsGR), plot.fit = FALSE)
enrichRes <- goseq(pwf, gene2cat = gata3Cons2Feature)
enrichRes %>%
  mutate(
    Percentage = percent(numDEInCat / numInCat),
    over_represented_pvalue = sprintf("%.3e", p.adjust(over_represented_pvalue, "bonf"))
  ) %>%
  dplyr::select(
    Feature = category,
    adjP = over_represented_pvalue,
    `Number Mapped in Group Peaks` = numDEInCat,
    `Consensus Peaks Mapped` = numInCat,
    `Percentage Mapped` = Percentage
  ) %>%
  pander(
    caption = "Assessment of enrichment for mappings of DHT-responsive GATA3 peaks to genomic features defined by H3K27Ac data. **These numbers need to be checked against those above!!!**"
  )
```


## Comparison between Promoters and Gene Expression

### Differentially Expressed Genes {.tabset}

Of the `r comma(nrow(rnaseq))` genes which were detectable in the corresponding RNA-Seq analysis, peaks corresponding to a gain of GATA3 under DHT treatment were found in the promoters of `r chipGR %>% join_overlap_intersect(h3k27ac) %>% subset(class == "Promoter") %>% join_overlap_intersect(geneRanges) %>% subset(gene %in% rnaseq$gene_id) %>% mcols() %>% .[["gene"]] %>% unique() %>% length()` genes.

```{r tabDEWithGATA3inPromoter}
chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  mcols() %>%
  as.data.frame() %>%
  as_tibble() %>%
  dplyr::rename(`H3K27Ac Group` = group) %>%
  dplyr::select(-peak_id.y) %>%
  chop(c(peak_id.x, feature_id)) %>%
  inner_join(rnaseq, by = c("gene" = "gene_id", "symbol" = "gene_name")) %>%
  arrange(PValue) %>%
  dplyr::filter(DE) %>%
  dplyr::mutate(
    `Peaks In Promoter` = vapply(peak_id.x, length, integer(1)),
    PValue = sprintf("%.2e", PValue),
    FDR = case_when(
      FDR > 0.001 ~ sprintf("%.3f", FDR),
      FDR <= 0.001 ~ sprintf("%.3e", FDR)
    )
  ) %>%
  dplyr::select(ID = gene, Name = symbol, `Peaks In Promoter`, `H3K27Ac Group`,  logFC, logCPM, PValue, FDR) %>%
  pander(
    justify = "llrlrrrr",
    caption = "
    RNA-Seq results for DHT-responsive GATA3 peaks which map to promoters. 
    Only Differentially expressed genes are shown, to an FDR of 5%. 
    All but one of these genes was up-regulated, however, given that most DE genes were up-regulated this was not statistically significant.
    ",
    emphasize.strong.rows = which(str_detect(.$`H3K27Ac Group`, "DHT"))
  )
```

```{r fetEnrichedPromoters}
rnaseq %>%
  mutate(
    gata3PeakInProm = gene_id %in% (
      chipGR %>%
        join_overlap_intersect(h3k27ac) %>%
        subset(class == "Promoter") %>%
        join_overlap_intersect(geneRanges) %>%
        mcols() %>%
        as.data.frame() %>% 
        pull("gene")
    ),
    DE = FDR < 0.05
  ) %>%
  group_by(DE, gata3PeakInProm) %>%
  tally() %>%
  pivot_wider(names_from = DE, values_from = n) %>%
  with(
    cbind(`FALSE`, `TRUE`)
  ) %>%
  fisher.test() %>%
  pander(
    caption = "Fisher's Exact Test providing weak support for enrichment of GATA3 gained peaks within the promoters of DE genes, using an FDR of 0.05 to consider a gene as DE"
  )
```

```{r fetForUpVsDown_GATA3, eval=FALSE, echo=FALSE}
# Check to see if there is enrichment for up-regulated genes in those with GATA3 peaks in the promoter. There isn't
rnaseq %>%
  mutate(
    gata3PeakInProm = gene_id %in% (
      chipGR %>%
        join_overlap_intersect(h3k27ac) %>%
        subset(class == "Promoter") %>%
        join_overlap_intersect(geneRanges) %>%
        mcols() %>%
        as.data.frame() %>% 
        pull("gene")
    ),
    DE = FDR < 0.05,
    Direction = case_when(logFC > 0 ~ "Up", logFC < 0 ~ "Down")
  ) %>%
  dplyr::filter(DE) %>%
  group_by(Direction, gata3PeakInProm) %>%
  tally() %>%
  with(
    matrix(n, nrow = 2, byrow = TRUE, dimnames = list(c("Down", "Up"), c("No GATA3","GATA3")))
  ) %>%
  fisher.test()
```

#### *ZBTB16*

The gene ZBTB16 has a gain of GATA3 in it's promoter under DHT treatment, as well as the gain of an H3K27Ac peak under DHT treatment.
It is also one of the most significantly up regulated genes in response to DHT.
Whilst an interaction between the ZBTB16 promoter and a downstream region was detected, this region did not interact with any other GATA peaks and appeared inconclusive.

However, it is clear that there are:

- Synchronous gains of both GATA3 and AR in the upstream enhancer, along with increased H3K27Ac enhancer marks for two of these three enhancers
- Displaced gains of AR and GATA3 in the upstream promoter, but with no increase in H3K27Ac signal
- Increased GATA3 without AR in the direct promoter, along with increased H3K27Ac signal in the promoter

```{r plotZBTB16, fig.cap = "*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. The rectangular box highlights the relevant GATA3 site*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "ZBTB16")
plotRegion(gr, zoom = 1000, gm = transModels, shift = 1e5)
```

```{r plotZBTB16_zoom, fig.cap = "*Zoomed in version of the above plot.*"}
plotRegion(gr, zoom = 150, gm = transModels, shift = -5e3, fs = 10)
```

#### *RANBP3L*

The most highly ranked DE gene with an AR-independent GATA3 gain under DHT was *RANBP3L*.
Given that we have GATA3 already in the promoter and AR is gained in the promoter, this may be a misclassification.
Is it worth considering a coverage threshold to consider a gain?
Is it worth performing a scan of nearby peaks to see if there are any AR gains at other peaks nearby?

```{r plotRANBP3L, fig.cap = "*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. The rectangular box highlights the relevant GATA3 site*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "RANBP3L")
plotRegion(gr, zoom = 240, gm = transModels, shift = -2.5e4)
```

#### *EHF*

The second most highly ranked DE gene with an AR-independent GATA3 gain under DHT was *EHF*.
There was a discrepancy with the peak assignment to group, with four peaks being considered as GATA3-gained in DHT only, but coverage was uncompelling.
These may need to be rethought.


```{r plotEHF, fig.cap = "*Zoomed view of EHF showing the 4 peaks assigned as AR-independent DHT-responsive. The first was clearly detected in GATA3_E2 as well as GATA3_ E2+DHT, whilst the second one was corectly annotated.*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "EHF")
plotRegion(gr[1], zoom = 20, gm = transModels)
plotRegion(gr[2], zoom = 50, gm = transModels)
plotRegion(gr[3], zoom = 50, gm = transModels)
plotRegion(gr[4], zoom = 100, gm = transModels)
```

However, the interactions detected connect this gene with *ELF5* and *APIP*, both of which were detected as expressed in the RNA-Seq data.
*ELF5* appears to be actively expressed independently of AR & GATA3, whilst *APIP* may be of interest.

```{r plotEHFExpanded, fig.cap = "*Expanded view of the above showing HiC contact locations*"}
plotRegion(range(gr), zoom = 20, gm = transModels, shift = 7e4, fs = 10)
```

#### *GJA1*

This peak again appears misclassified, as it wasn't found in the GATA3 E2+DHT bed file.
However, there appears to be be increased activity at the promoter.
The HiC interaction connects to a region with no defined genes, promoters or enhancers.

```{r plotGJA1, fig.cap = "*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. The rectangular box highlights the relevant GATA3 site*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "GJA1")
plotRegion(gr, zoom = 20, gm = transModels, shift = 3e3)
```

#### *KDM4B*

These peaks again appear potentially misclassified.
However, there appears to be be increased activity at the promoter.
The HiC interaction connects to a region with no defined genes, promoters or enhancers.

```{r plotKDM4B, fig.cap = "*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. The rectangular box highlights the relevant GATA3 site*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "KDM4B")
plotRegion(gr[1], zoom = 80, gm = transModels)
plotRegion(gr[2], zoom = 200, gm = transModels, shift = 6e3)
```

#### *SASH1*

Another peak which is not very compelling

```{r plotSASH1, fig.cap = "*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. The rectangular box highlights the relevant GATA3 site*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "SASH1")
plotRegion(gr[1], zoom = 50, gm = transModels)
```

#### *DIPK2A*

Another peak which is not very compelling

```{r plotDIPK2A, fig.cap = "*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. The rectangular box highlights the relevant GATA3 site*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "DIPK2A")
plotRegion(gr[1], zoom = 50, gm = transModels)
```

#### *TBC1D16*

Another peak which is not very compelling

```{r plotTBC1D16, fig.cap = "*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. The rectangular box highlights the relevant GATA3 site*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "TBC1D16")
plotRegion(gr[1], zoom = 50, gm = transModels)
plotRegion(gr[1], zoom = 400, gm = transModels)
```


#### *GREB1*


```{r plotGREB1, fig.cap = "*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. The rectangular box highlights the relevant GATA3 site*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "GREB1")
plotRegion(range(gr), zoom = 2.5, shift = 4e4, gm = transModels)
```

These `r length(gr)` peaks again appear un compelling.

```{r plotGREB1_peaks}
plotRegion(gr[1], zoom = 100, gm = transModels, shift = 9e3)
plotRegion(gr[2], zoom = 200, gm = transModels)
plotRegion(gr[3], zoom = 100, gm = transModels)
plotRegion(gr[4], zoom = 100, gm = transModels)
plotRegion(gr[5], zoom = 50, gm = transModels)
```

#### *FKBP5*

I'd almost buy this one apart from the really weak signal.
Do we have one transcript responding to AR, whilst there is another responding to GATA3 without AR?

```{r plotFKBP5, fig.cap = "*Overlay of HiC Interactions, GATA3 binding sites and H3K27Ac marks. For GATA3 peaks, Promoters, Enhancers and Super Enhancers, red indicates a gain under DHT, grey represents common sites across treatments and blue indicates a loss under DHT. Expressed genes are displayed in green with all transcripts showing. Undetectable genes from the RNA Seq dataset are condensed into a single representative transcript in the bottom panel. The rectangular box highlights the relevant GATA3 site*"}
gr <- chipGR %>%
  join_overlap_intersect(h3k27ac) %>%
  subset(class == "Promoter") %>%
  join_overlap_intersect(geneRanges) %>%
  subset(symbol == "FKBP5")
plotRegion(gr[1], zoom = 80, gm = transModels)
plotRegion(gr[1], zoom = 700, gm = transModels, shift = -4e4)
```
