---
title: "Description of Workflow"
author: "Stephen Pederson<br>Dame Roma Mitchell Cancer Research Laboratories<br>Adelaide Medical School<br>University of Adelaide"
date: "`r format(Sys.Date(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center"
)
```

```{r packages}
library(tidyverse)
library(DiagrammeR)
library(pander)
library(yaml)
library(spBioUtils)
```

```{r config}
config <- here::here("config/config.yml") %>%
  read_yaml()
```


# Pre-Processing Workflow

## Workflow Outline


This is an analysis of RNA-Seq data with the data processing workflow derived from https://github.com/steveped/snakemake_rnaseq_se.
The pre-processing workflow as implemented in this analysis is as follows:

```{r plot_workflow, fig.cap = "*Summary of workflow. Node names indicate snakemake rules. Blue nodes indicate preparation of the genomic reference, green nodes indicate preparation of the sequencing data, red nodes indicate the HiC-Pro workflow, whlist the remaining purple nodes indicate the detection of significant interactions*"}
here::here("output/rulegraph.dot") %>%
  readLines() %>%
  rm_dot_node() %>%
  change_node_colour("raw|adapter|fastqc", "green") %>%
  change_node_colour("hicpro|maps", "red") %>%
  change_node_colour("ref|chrom|frag|bowtie", "blue") %>%
  change_node_colour("maxhic|merge_", rgb(0.5, 0.4, 0.7)) %>%
  grViz()
```

## Parameter Description

Parameters set for the complete workflow are:

```{r printYAML, echo=FALSE, results='asis'}
pander(config)
```

# Data Description

```{r samples}
samples <- config$samples %>%
  here::here() %>%
  read_tsv() %>%
  mutate_if(
    function(x){length(unique(x)) < length(x)},
    as.factor
  )
n_fact <- sum(vapply(samples, is.factor, logical(1)))
```

