---
title: "Interactions Setup"
author: "Stephen Pederson<br>Dame Roma Mitchell Cancer Research Laboratories<br>Adelaide Medical School<br>University of Adelaide"
date: "`r format(Sys.Date(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center"
)
```

```{r packages}
library(tidyverse)
library(magrittr)
library(yaml)
library(rtracklayer)
library(InteractionSet)
library(vroom)
library(glue)
library(scales)
library(pander)
library(cowplot)
```

```{r setOpts}
theme_set(theme_bw())
```


```{r config}
config <- here::here("config/config.yml") %>%
  read_yaml()
```

# Introduction

The output from MaxHiC is extremely large and for some bin-sizes, near impossible to wrangle on a standard laptop.
In this workflow, each set of outputs containing the **cis** interactions will be loaded and reduced to only the *significant* interactions.
These far smaller sets of interactions are easily managed in a standard compute environment.
Analyses were run for genomic bins of `r pander(as.numeric(str_split(config$hicpro$bin_size, pattern = " ")[[1]])/1e3)`kb.

```{r}
allGR <- read_rds(
  here::here("output/allGR.rds")
)
sq <- seqinfo(allGR)
rm(allGR)
```


# Data Preparation {.tabset}

## 10kb Bins

```{r sz10}
sz <- 10000
```

```{r bins10}
bins <- import.bed(
  glue("data/hic/hic_results/matrix/merged/raw/{sz}/merged_{sz}_abs.bed"),
  seqinfo = sq
)
```

```{r cis_int_10}
cis_int <- vroom(
  glue("output/MaxHiC/merged/{sz}/cis_interactions.txt.gz"),
  col_types = "dddd-d--"
)
fdr <- p.adjust(exp(-cis_int$neg_ln_p_val), "fdr")
keepInt <- fdr < 0.05
```

By default, any bin pairs with no observed interactions were excluded from the Max-HiC output.
Using `r sz/1e3`kb bins, a total of `r comma(nrow(cis_int))` non-zero interactions were observed between bin pairs.
The median observed interaction count was `r median(cis_int$observed_interactions)`, whilst the maximum was `r max(cis_int$observed_interactions)`.

By default, the FDR provided by Max-HiC is incorrect and was ignored.
The MaxHiC output was defined as a `GInteractions` object, with correct FDR added and interactions filtered for those with an FDR < 0.05.

```{r gi10, results='hide'}
gc()
gi <- GInteractions(
  anchor1 = bins[as.integer(cis_int$bin1ID[keepInt])],
  anchor2 = bins[as.integer(cis_int$bin2ID[keepInt])],
  counts = as.integer(cis_int$observed_interactions[keepInt]),
  exp_interactions = cis_int$exp_interactions[keepInt],
  p = exp(-cis_int$neg_ln_p_val)[keepInt],
  fdr = fdr[keepInt]
)
gi$distance <- pairdist(gi)
gi$logObsExp <- log2(gi$counts / gi$exp_interactions)
seqinfo(gi) <- sq
```

```{r export_gi10, results='hide'}
write_rds(
  gi,
  here::here(
    "output",
    glue("gi_{sz}.rds")
  ),
  compress = "gz"
)
gc()
```


After filtering interactions to only those considered significant:

- `r comma(length(gi))` bin-pair interactions were retained.
- The observed interactions between bins ranged between `r pander(range(gi$counts))`
- The log~2~ ratio of of observed to expected counts ranged between `r pander(round(range(gi$logObsExp), 2))`. These values represent enrichment for interactions well above that expected by the background model.
- The median interaction distance was `r round(median(gi$distance)/1e3, 1)`kb
- The longest interaction was between bins separated by `r round(max(gi$distance)/1e6, 1)`Mb

```{r plotInt10, fig.height=6, fig.width = 9, fig.cap = glue("*Genomic Distribution of significant interactions using {sz/1e3}kb bins*")}
gi %>%
  anchors("first") %>%
  seqnames() %>%
  table() %>%
  enframe(
    name = "chr",
    value = "count"
  ) %>%
  mutate(
    chr =factor(chr, levels = seqlevels(sq)),
    count = as.integer(count)
  ) %>%
  ggplot(
    aes(chr, count)
  ) +
  geom_col() +
  labs(
    x = "Chromosome",
    y = glue("Total Significant Cis Interactions ({sz/1e3}kb bins)")
  ) +
  scale_y_continuous(expand = expansion(c(0, 0.05)))
```

```{r plotIntDist10,fig.height=6, fig.width = 9, fig.cap = glue("*Relationship between counts and distance using {sz/1e3}kb bins*")}
a <- mcols(gi) %>% 
  as.data.frame %>% 
  ggplot(aes(distance/1e3, counts)) + 
  geom_point() + 
  scale_x_log10(labels = comma) +
  labs(
    x = "Distance between bins (kb)",
    y = "Reads Spanning Bins"
  )
b <- mcols(gi) %>%
  as.data.frame %>% 
  ggplot(aes(counts, stat(density))) + 
  geom_density() + 
  coord_flip() + 
  labs(x = c(), y = "") +
  theme(
    axis.text = element_blank(), 
    panel.grid.major.x =  element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank()
    )
c <- mcols(gi) %>%
  as.data.frame %>% 
  ggplot(aes(distance/1e3, stat(density))) + 
  geom_density() + 
  scale_x_log10() +
  labs(x = c(), y = "") +
  theme(
    panel.grid.major.y = element_blank(),
    panel.grid.minor.y = element_blank(),
    axis.text = element_blank(),
    axis.ticks.y = element_blank(), 
    axis.ticks.x = element_blank()
  )
plot_grid(
  plot_grid(ngsReports:::.emptyPlot(""), c, ngsReports:::.emptyPlot(""), rel_widths = c(0.02, 0.896, 0.102), nrow = 1),
  plot_grid(a, b, rel_widths = c(0.9, 0.1), align = "h"),
  rel_heights = c(0.1, 0.9), nrow = 2, align = "v", axis = "r"
)
```


```{r cleanup10, results='hide'}
rm(gi)
rm(cis_int)
rm(fdr)
rm(keepInt)
gc()
```


```{r sz20}
sz <- 20000
```

## `r sz/1e3`kb Bins

```{r bins20}
bins <- import.bed(
  glue("data/hic/hic_results/matrix/merged/raw/{sz}/merged_{sz}_abs.bed"),
  seqinfo = sq
)
```

```{r cis_int_20}
cis_int <- vroom(
  glue("output/MaxHiC/merged/{sz}/cis_interactions.txt.gz"),
  col_types = "dddd-d--"
)
fdr <- p.adjust(exp(-cis_int$neg_ln_p_val), "fdr")
keepInt <- fdr < 0.05
```

By default, any bin pairs with no observed interactions were excluded from the Max-HiC output.
Using `r sz/1e3`kb bins, a total of `r comma(nrow(cis_int))` non-zero interactions were observed between bin pairs.
The median observed interaction count was `r median(cis_int$observed_interactions)`, whilst the maximum was `r max(cis_int$observed_interactions)`.

By default, the FDR provided by Max-HiC is incorrect and was ignored.
The MaxHiC output was defined as a `GInteractions` object, with correct FDR added and interactions filtered for those with an FDR < 0.05.

```{r gi20, results='hide'}
gc()
gi <- GInteractions(
  anchor1 = bins[as.integer(cis_int$bin1ID[keepInt])],
  anchor2 = bins[as.integer(cis_int$bin2ID[keepInt])],
  counts = as.integer(cis_int$observed_interactions[keepInt]),
  exp_interactions = cis_int$exp_interactions[keepInt],
  p = exp(-cis_int$neg_ln_p_val)[keepInt],
  fdr = fdr[keepInt]
)
gi$distance <- pairdist(gi)
gi$logObsExp <- log2(gi$counts / gi$exp_interactions)
seqinfo(gi) <- sq
```

```{r export_gi20, results='hide'}
write_rds(
  gi,
  here::here(
    "output",
    glue("gi_{sz}.rds")
  ),
  compress = "gz"
)
gc()
```


After filtering interactions to only those considered significant:

- `r comma(length(gi))` bin-pair interactions were retained.
- The observed interactions between bins ranged between `r pander(range(gi$counts))`
- The log~2~ ratio of of observed to expected counts ranged between `r pander(round(range(gi$logObsExp), 2))`. These values represent enrichment for interactions well above that expected by the background model.
- The median interaction distance was `r round(median(gi$distance)/1e3, 1)`kb
- The longest interaction was between bins separated by `r round(max(gi$distance)/1e6, 1)`Mb

```{r plotInt20, fig.height=6, fig.width = 9, fig.cap = glue("*Genomic Distribution of significant interactions using {sz/1e3}kb bins*")}
gi %>%
  anchors("first") %>%
  seqnames() %>%
  table() %>%
  enframe(
    name = "chr",
    value = "count"
  ) %>%
  mutate(
    chr =factor(chr, levels = seqlevels(sq)),
    count = as.integer(count)
  ) %>%
  ggplot(
    aes(chr, count)
  ) +
  geom_col() +
  labs(
    x = "Chromosome",
    y = glue("Total Significant Cis Interactions ({sz/1e3}kb bins)")
  ) +
  scale_y_continuous(expand = expansion(c(0, 0.05)))
```

```{r plotIntDist20,fig.height=6, fig.width = 9, fig.cap = glue("*Relationship between counts and distance using {sz/1e3}kb bins*")}
a <- mcols(gi) %>% 
  as.data.frame %>% 
  ggplot(aes(distance/1e3, counts)) + 
  geom_point() + 
  scale_x_log10(labels = comma) +
  labs(
    x = "Distance between bins (kb)",
    y = "Reads Spanning Bins"
  )
b <- mcols(gi) %>%
  as.data.frame %>% 
  ggplot(aes(counts, stat(density))) + 
  geom_density() + 
  coord_flip() + 
  labs(x = c(), y = "") +
  theme(
    axis.text = element_blank(), 
    panel.grid.major.x =  element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank()
    )
c <- mcols(gi) %>%
  as.data.frame %>% 
  ggplot(aes(distance/1e3, stat(density))) + 
  geom_density() + 
  scale_x_log10() +
  labs(x = c(), y = "") +
  theme(
    panel.grid.major.y = element_blank(),
    panel.grid.minor.y = element_blank(),
    axis.text = element_blank(),
    axis.ticks.y = element_blank(), 
    axis.ticks.x = element_blank()
  )
plot_grid(
  plot_grid(ngsReports:::.emptyPlot(""), c, ngsReports:::.emptyPlot(""), rel_widths = c(0.02, 0.896, 0.102), nrow = 1),
  plot_grid(a, b, rel_widths = c(0.9, 0.1), align = "h"),
  rel_heights = c(0.1, 0.9), nrow = 2, align = "v", axis = "r"
)
```


```{r cleanup20, results='hide'}
rm(gi)
rm(cis_int)
rm(fdr)
rm(keepInt)
gc()
```


```{r sz40}
sz <- 40000
```

## `r sz/1e3`kb Bins

```{r bins40}
bins <- import.bed(
  glue("data/hic/hic_results/matrix/merged/raw/{sz}/merged_{sz}_abs.bed"),
  seqinfo = sq
)
```

```{r cis_int_40}
cis_int <- vroom(
  glue("output/MaxHiC/merged/{sz}/cis_interactions.txt.gz"),
  col_types = "dddd-d--"
)
fdr <- p.adjust(exp(-cis_int$neg_ln_p_val), "fdr")
keepInt <- fdr < 0.05
```

By default, any bin pairs with no observed interactions were excluded from the Max-HiC output.
Using `r sz/1e3`kb bins, a total of `r comma(nrow(cis_int))` non-zero interactions were observed between bin pairs.
The median observed interaction count was `r median(cis_int$observed_interactions)`, whilst the maximum was `r max(cis_int$observed_interactions)`.

By default, the FDR provided by Max-HiC is incorrect and was ignored.
The MaxHiC output was defined as a `GInteractions` object, with correct FDR added and interactions filtered for those with an FDR < 0.05.

```{r gi40, results='hide'}
gc()
gi <- GInteractions(
  anchor1 = bins[as.integer(cis_int$bin1ID[keepInt])],
  anchor2 = bins[as.integer(cis_int$bin2ID[keepInt])],
  counts = as.integer(cis_int$observed_interactions[keepInt]),
  exp_interactions = cis_int$exp_interactions[keepInt],
  p = exp(-cis_int$neg_ln_p_val)[keepInt],
  fdr = fdr[keepInt]
)
gi$distance <- pairdist(gi)
gi$logObsExp <- log2(gi$counts / gi$exp_interactions)
seqinfo(gi) <- sq
```

```{r export_gi40, results='hide'}
write_rds(
  gi,
  here::here(
    "output",
    glue("gi_{sz}.rds")
  ),
  compress = "gz"
)
gc()
```


After filtering interactions to only those considered significant:

- `r comma(length(gi))` bin-pair interactions were retained.
- The observed interactions between bins ranged between `r pander(range(gi$counts))`
- The log~2~ ratio of of observed to expected counts ranged between `r pander(round(range(gi$logObsExp), 2))`. These values represent enrichment for interactions well above that expected by the background model.
- The median interaction distance was `r round(median(gi$distance)/1e3, 1)`kb
- The longest interaction was between bins separated by `r round(max(gi$distance)/1e6, 1)`Mb

```{r plotInt40, fig.height=6, fig.width = 9, fig.cap = glue("*Genomic Distribution of significant interactions using {sz/1e3}kb bins*")}
gi %>%
  anchors("first") %>%
  seqnames() %>%
  table() %>%
  enframe(
    name = "chr",
    value = "count"
  ) %>%
  mutate(
    chr =factor(chr, levels = seqlevels(sq)),
    count = as.integer(count)
  ) %>%
  ggplot(
    aes(chr, count)
  ) +
  geom_col() +
  labs(
    x = "Chromosome",
    y = glue("Total Significant Cis Interactions ({sz/1e3}kb bins)")
  ) +
  scale_y_continuous(expand = expansion(c(0, 0.05)))
```

```{r plotIntDist40,fig.height=6, fig.width = 9, fig.cap = glue("*Relationship between counts and distance using {sz/1e3}kb bins*")}
a <- mcols(gi) %>% 
  as.data.frame %>% 
  ggplot(aes(distance/1e3, counts)) + 
  geom_point() + 
  scale_x_log10(labels = comma) +
  labs(
    x = "Distance between bins (kb)",
    y = "Reads Spanning Bins"
  )
b <- mcols(gi) %>%
  as.data.frame %>% 
  ggplot(aes(counts, stat(density))) + 
  geom_density() + 
  coord_flip() + 
  labs(x = c(), y = "") +
  theme(
    axis.text = element_blank(), 
    panel.grid.major.x =  element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank()
    )
c <- mcols(gi) %>%
  as.data.frame %>% 
  ggplot(aes(distance/1e3, stat(density))) + 
  geom_density() + 
  scale_x_log10() +
  labs(x = c(), y = "") +
  theme(
    panel.grid.major.y = element_blank(),
    panel.grid.minor.y = element_blank(),
    axis.text = element_blank(),
    axis.ticks.y = element_blank(), 
    axis.ticks.x = element_blank()
  )
plot_grid(
  plot_grid(ngsReports:::.emptyPlot(""), c, ngsReports:::.emptyPlot(""), rel_widths = c(0.02, 0.896, 0.102), nrow = 1),
  plot_grid(a, b, rel_widths = c(0.9, 0.1), align = "h"),
  rel_heights = c(0.1, 0.9), nrow = 2, align = "v", axis = "r"
)
```


```{r cleanup40, results='hide'}
rm(gi)
rm(cis_int)
rm(fdr)
rm(keepInt)
gc()
```


```{r sz100}
sz <- "100000"
```

## `r as.numeric(sz)/1e3`kb Bins

```{r bins100}
bins <- import.bed(
  glue("data/hic/hic_results/matrix/merged/raw/{sz}/merged_{sz}_abs.bed"),
  seqinfo = sq
)
```


```{r cis_int_100}
cis_int <- vroom(
  glue("output/MaxHiC/merged/{sz}/cis_interactions.txt.gz"),
  col_types = "dddd-d--"
)
```

By default, any bin pairs with no observed interactions were excluded from the Max-HiC output.
Using `r as.numeric(sz)/1e3`kb bins, a total of `r comma(nrow(cis_int))` non-zero interactions were observed between bin pairs.
The median observed interaction count was `r median(cis_int$observed_interactions)`, whilst the maximum was `r max(cis_int$observed_interactions)`.

By default, the FDR provided by Max-HiC is incorrect and was ignored.
The MaxHiC output was defined as a `GInteractions` object, with correct FDR added and interactions filtered for those with an FDR < 0.05

```{r gi100}
gi <- GInteractions(
  anchor1 = bins[as.integer(cis_int$bin1ID)],
  anchor2 = bins[as.integer(cis_int$bin2ID)],
  counts = as.integer(cis_int$observed_interactions),
  exp_interactions = cis_int$exp_interactions,
  p = exp(-cis_int$neg_ln_p_val)
)
seqinfo(gi) <- sq
gi$fdr <- p.adjust(gi$p, "fdr")
gi$distance <- pairdist(gi)
gi$logObsExp <- log2(gi$counts / gi$exp_interactions)
```

```{r export_gi100, results='hide'}
gi <- subset(gi, fdr < 0.05)
write_rds(
  gi,
  here::here(
    "output",
    glue("gi_{sz}.rds")
  ),
  compress = "gz"
)
gc()
```

After filtering interactions to only those considered significant:

- `r comma(length(gi))` bin-pair interactions were retained.
- The observed interactions between bins ranged between `r pander(range(gi$counts))`
- The log~2~ ratio of of observed to expected counts ranged between `r pander(round(range(gi$logObsExp), 2))`. These values represent enrichment for interactions well above that expected by the background model.

```{r plotIn100, fig.height=6, fig.width = 9, fig.cap = glue("*Distribution of significant interactions using {as.numeric(sz)/1e3}kb bins*")}
gi %>%
  anchors("first") %>%
  seqnames() %>%
  table() %>%
  enframe(
    name = "chr",
    value = "count"
  ) %>%
  mutate(
    chr =factor(chr, levels = seqlevels(sq)),
    count = as.integer(count)
  ) %>%
  ggplot(
    aes(chr, count)
  ) +
  geom_col() +
  labs(
    x = "Chromosome",
    y = glue("Total Significant Cis Interactions ({as.numeric(sz)/1e3}kb bins)")
  ) +
  scale_y_continuous(expand = expansion(c(0, 0.05)))
```

```{r cleanup100, results='hide'}
rm(gi)
rm(cis_int)
gc()
```

